// @pulumi/aws@7.7.0/connect/phoneNumber.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PhoneNumber = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Amazon Connect Phone Number resource. For more information see
 * [Amazon Connect: Getting Started](https://docs.aws.amazon.com/connect/latest/adminguide/amazon-connect-get-started.html)
 *
 * ## Example Usage
 *
 * ### Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.connect.PhoneNumber("example", {
 *     targetArn: exampleAwsConnectInstance.arn,
 *     countryCode: "US",
 *     type: "DID",
 *     tags: {
 *         hello: "world",
 *     },
 * });
 * ```
 *
 * ### Description
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.connect.PhoneNumber("example", {
 *     targetArn: exampleAwsConnectInstance.arn,
 *     countryCode: "US",
 *     type: "DID",
 *     description: "example description",
 * });
 * ```
 *
 * ### Prefix to filter phone numbers
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.connect.PhoneNumber("example", {
 *     targetArn: exampleAwsConnectInstance.arn,
 *     countryCode: "US",
 *     type: "DID",
 *     prefix: "+18005",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Amazon Connect Phone Numbers using its `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:connect/phoneNumber:PhoneNumber example 12345678-abcd-1234-efgh-9876543210ab
 * ```
 */
class PhoneNumber extends pulumi.CustomResource {
    /**
     * Get an existing PhoneNumber resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PhoneNumber(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of PhoneNumber.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PhoneNumber.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["countryCode"] = state?.countryCode;
            resourceInputs["description"] = state?.description;
            resourceInputs["phoneNumber"] = state?.phoneNumber;
            resourceInputs["prefix"] = state?.prefix;
            resourceInputs["region"] = state?.region;
            resourceInputs["statuses"] = state?.statuses;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["targetArn"] = state?.targetArn;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            if (args?.countryCode === undefined && !opts.urn) {
                throw new Error("Missing required property 'countryCode'");
            }
            if (args?.targetArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetArn'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["countryCode"] = args?.countryCode;
            resourceInputs["description"] = args?.description;
            resourceInputs["prefix"] = args?.prefix;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["targetArn"] = args?.targetArn;
            resourceInputs["type"] = args?.type;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["phoneNumber"] = undefined /*out*/;
            resourceInputs["statuses"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PhoneNumber.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PhoneNumber = PhoneNumber;
/** @internal */
PhoneNumber.__pulumiType = 'aws:connect/phoneNumber:PhoneNumber';
