// @pulumi/aws@7.7.0/emrserverless/application.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an EMR Serverless Application.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.emrserverless.Application("example", {
 *     name: "example",
 *     releaseLabel: "emr-6.6.0",
 *     type: "hive",
 * });
 * ```
 *
 * ### Initial Capacity Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.emrserverless.Application("example", {
 *     name: "example",
 *     releaseLabel: "emr-6.6.0",
 *     type: "hive",
 *     initialCapacities: [{
 *         initialCapacityType: "HiveDriver",
 *         initialCapacityConfig: {
 *             workerCount: 1,
 *             workerConfiguration: {
 *                 cpu: "2 vCPU",
 *                 memory: "10 GB",
 *             },
 *         },
 *     }],
 * });
 * ```
 *
 * ### Maximum Capacity Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.emrserverless.Application("example", {
 *     name: "example",
 *     releaseLabel: "emr-6.6.0",
 *     type: "hive",
 *     maximumCapacity: {
 *         cpu: "2 vCPU",
 *         memory: "10 GB",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import EMR Severless applications using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:emrserverless/application:Application example id
 * ```
 */
class Application extends pulumi.CustomResource {
    /**
     * Get an existing Application resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Application(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Application.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Application.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["architecture"] = state?.architecture;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["autoStartConfiguration"] = state?.autoStartConfiguration;
            resourceInputs["autoStopConfiguration"] = state?.autoStopConfiguration;
            resourceInputs["imageConfiguration"] = state?.imageConfiguration;
            resourceInputs["initialCapacities"] = state?.initialCapacities;
            resourceInputs["interactiveConfiguration"] = state?.interactiveConfiguration;
            resourceInputs["maximumCapacity"] = state?.maximumCapacity;
            resourceInputs["name"] = state?.name;
            resourceInputs["networkConfiguration"] = state?.networkConfiguration;
            resourceInputs["region"] = state?.region;
            resourceInputs["releaseLabel"] = state?.releaseLabel;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            if (args?.releaseLabel === undefined && !opts.urn) {
                throw new Error("Missing required property 'releaseLabel'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["architecture"] = args?.architecture;
            resourceInputs["autoStartConfiguration"] = args?.autoStartConfiguration;
            resourceInputs["autoStopConfiguration"] = args?.autoStopConfiguration;
            resourceInputs["imageConfiguration"] = args?.imageConfiguration;
            resourceInputs["initialCapacities"] = args?.initialCapacities;
            resourceInputs["interactiveConfiguration"] = args?.interactiveConfiguration;
            resourceInputs["maximumCapacity"] = args?.maximumCapacity;
            resourceInputs["name"] = args?.name;
            resourceInputs["networkConfiguration"] = args?.networkConfiguration;
            resourceInputs["region"] = args?.region;
            resourceInputs["releaseLabel"] = args?.releaseLabel;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["type"] = args?.type;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Application.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Application = Application;
/** @internal */
Application.__pulumiType = 'aws:emrserverless/application:Application';
