// @pulumi/aws@7.7.0/datasync/efsLocation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EfsLocation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an AWS DataSync EFS Location.
 *
 * > **NOTE:** The EFS File System must have a mounted EFS Mount Target before creating this resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.datasync.EfsLocation("example", {
 *     efsFileSystemArn: exampleAwsEfsMountTarget.fileSystemArn,
 *     ec2Config: {
 *         securityGroupArns: [exampleAwsSecurityGroup.arn],
 *         subnetArn: exampleAwsSubnet.arn,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_datasync_location_efs` using the DataSync Task Amazon Resource Name (ARN). For example:
 *
 * ```sh
 * $ pulumi import aws:datasync/efsLocation:EfsLocation example arn:aws:datasync:us-east-1:123456789012:location/loc-12345678901234567
 * ```
 */
class EfsLocation extends pulumi.CustomResource {
    /**
     * Get an existing EfsLocation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EfsLocation(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EfsLocation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EfsLocation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessPointArn"] = state?.accessPointArn;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["ec2Config"] = state?.ec2Config;
            resourceInputs["efsFileSystemArn"] = state?.efsFileSystemArn;
            resourceInputs["fileSystemAccessRoleArn"] = state?.fileSystemAccessRoleArn;
            resourceInputs["inTransitEncryption"] = state?.inTransitEncryption;
            resourceInputs["region"] = state?.region;
            resourceInputs["subdirectory"] = state?.subdirectory;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["uri"] = state?.uri;
        }
        else {
            const args = argsOrState;
            if (args?.ec2Config === undefined && !opts.urn) {
                throw new Error("Missing required property 'ec2Config'");
            }
            if (args?.efsFileSystemArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'efsFileSystemArn'");
            }
            resourceInputs["accessPointArn"] = args?.accessPointArn;
            resourceInputs["ec2Config"] = args?.ec2Config;
            resourceInputs["efsFileSystemArn"] = args?.efsFileSystemArn;
            resourceInputs["fileSystemAccessRoleArn"] = args?.fileSystemAccessRoleArn;
            resourceInputs["inTransitEncryption"] = args?.inTransitEncryption;
            resourceInputs["region"] = args?.region;
            resourceInputs["subdirectory"] = args?.subdirectory;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["uri"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EfsLocation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EfsLocation = EfsLocation;
/** @internal */
EfsLocation.__pulumiType = 'aws:datasync/efsLocation:EfsLocation';
