// @pulumi/aws@7.7.0/datasync/locationFsxWindows.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocationFsxWindows = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an AWS DataSync FSx Windows Location.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.datasync.LocationFsxWindows("example", {
 *     fsxFilesystemArn: exampleAwsFsxWindowsFileSystem.arn,
 *     user: "SomeUser",
 *     password: "SuperSecretPassw0rd",
 *     securityGroupArns: [exampleAwsSecurityGroup.arn],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_datasync_location_fsx_windows_file_system` using the `DataSync-ARN#FSx-Windows-ARN`. For example:
 *
 * ```sh
 * $ pulumi import aws:datasync/locationFsxWindows:LocationFsxWindows example arn:aws:datasync:us-west-2:123456789012:location/loc-12345678901234567#arn:aws:fsx:us-west-2:476956259333:file-system/fs-08e04cd442c1bb94a
 * ```
 */
class LocationFsxWindows extends pulumi.CustomResource {
    /**
     * Get an existing LocationFsxWindows resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LocationFsxWindows(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of LocationFsxWindows.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LocationFsxWindows.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["creationTime"] = state?.creationTime;
            resourceInputs["domain"] = state?.domain;
            resourceInputs["fsxFilesystemArn"] = state?.fsxFilesystemArn;
            resourceInputs["password"] = state?.password;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityGroupArns"] = state?.securityGroupArns;
            resourceInputs["subdirectory"] = state?.subdirectory;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["uri"] = state?.uri;
            resourceInputs["user"] = state?.user;
        }
        else {
            const args = argsOrState;
            if (args?.fsxFilesystemArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'fsxFilesystemArn'");
            }
            if (args?.password === undefined && !opts.urn) {
                throw new Error("Missing required property 'password'");
            }
            if (args?.securityGroupArns === undefined && !opts.urn) {
                throw new Error("Missing required property 'securityGroupArns'");
            }
            if (args?.user === undefined && !opts.urn) {
                throw new Error("Missing required property 'user'");
            }
            resourceInputs["domain"] = args?.domain;
            resourceInputs["fsxFilesystemArn"] = args?.fsxFilesystemArn;
            resourceInputs["password"] = args?.password ? pulumi.secret(args.password) : undefined;
            resourceInputs["region"] = args?.region;
            resourceInputs["securityGroupArns"] = args?.securityGroupArns;
            resourceInputs["subdirectory"] = args?.subdirectory;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["user"] = args?.user;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["creationTime"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["uri"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["password"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(LocationFsxWindows.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LocationFsxWindows = LocationFsxWindows;
/** @internal */
LocationFsxWindows.__pulumiType = 'aws:datasync/locationFsxWindows:LocationFsxWindows';
