// @pulumi/aws@7.7.0/datasync/s3location.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.S3Location = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an S3 Location within AWS DataSync.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.datasync.S3Location("example", {
 *     s3BucketArn: exampleAwsS3Bucket.arn,
 *     subdirectory: "/example/prefix",
 *     s3Config: {
 *         bucketAccessRoleArn: exampleAwsIamRole.arn,
 *     },
 * });
 * ```
 *
 * ### S3 Bucket on AWS Outposts
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const destination = new aws.datasync.S3Location("destination", {
 *     agentArns: [exampleAwsDatasyncAgent.arn],
 *     s3BucketArn: exampleAwsS3AccessPoint.arn,
 *     s3StorageClass: "OUTPOSTS",
 *     subdirectory: "/example/prefix",
 *     s3Config: {
 *         bucketAccessRoleArn: example.arn,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_datasync_location_s3` using the DataSync Task Amazon Resource Name (ARN). For example:
 *
 * ```sh
 * $ pulumi import aws:datasync/s3Location:S3Location example arn:aws:datasync:us-east-1:123456789012:location/loc-12345678901234567
 * ```
 */
class S3Location extends pulumi.CustomResource {
    /**
     * Get an existing S3Location resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new S3Location(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of S3Location.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === S3Location.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentArns"] = state?.agentArns;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["region"] = state?.region;
            resourceInputs["s3BucketArn"] = state?.s3BucketArn;
            resourceInputs["s3Config"] = state?.s3Config;
            resourceInputs["s3StorageClass"] = state?.s3StorageClass;
            resourceInputs["subdirectory"] = state?.subdirectory;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["uri"] = state?.uri;
        }
        else {
            const args = argsOrState;
            if (args?.s3BucketArn === undefined && !opts.urn) {
                throw new Error("Missing required property 's3BucketArn'");
            }
            if (args?.s3Config === undefined && !opts.urn) {
                throw new Error("Missing required property 's3Config'");
            }
            if (args?.subdirectory === undefined && !opts.urn) {
                throw new Error("Missing required property 'subdirectory'");
            }
            resourceInputs["agentArns"] = args?.agentArns;
            resourceInputs["region"] = args?.region;
            resourceInputs["s3BucketArn"] = args?.s3BucketArn;
            resourceInputs["s3Config"] = args?.s3Config;
            resourceInputs["s3StorageClass"] = args?.s3StorageClass;
            resourceInputs["subdirectory"] = args?.subdirectory;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["uri"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(S3Location.__pulumiType, name, resourceInputs, opts);
    }
}
exports.S3Location = S3Location;
/** @internal */
S3Location.__pulumiType = 'aws:datasync/s3Location:S3Location';
