// @pulumi/aws@7.7.0/pinpoint/smsChannel.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SmsChannel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Use the `aws.pinpoint.SmsChannel` resource to manage Pinpoint SMS Channels.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const app = new aws.pinpoint.App("app", {});
 * const sms = new aws.pinpoint.SmsChannel("sms", {applicationId: app.applicationId});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import the Pinpoint SMS Channel using the `application_id`. For example:
 *
 * ```sh
 * $ pulumi import aws:pinpoint/smsChannel:SmsChannel sms application-id
 * ```
 */
class SmsChannel extends pulumi.CustomResource {
    /**
     * Get an existing SmsChannel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SmsChannel(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of SmsChannel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SmsChannel.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["applicationId"] = state?.applicationId;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["promotionalMessagesPerSecond"] = state?.promotionalMessagesPerSecond;
            resourceInputs["region"] = state?.region;
            resourceInputs["senderId"] = state?.senderId;
            resourceInputs["shortCode"] = state?.shortCode;
            resourceInputs["transactionalMessagesPerSecond"] = state?.transactionalMessagesPerSecond;
        }
        else {
            const args = argsOrState;
            if (args?.applicationId === undefined && !opts.urn) {
                throw new Error("Missing required property 'applicationId'");
            }
            resourceInputs["applicationId"] = args?.applicationId;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["region"] = args?.region;
            resourceInputs["senderId"] = args?.senderId;
            resourceInputs["shortCode"] = args?.shortCode;
            resourceInputs["promotionalMessagesPerSecond"] = undefined /*out*/;
            resourceInputs["transactionalMessagesPerSecond"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SmsChannel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SmsChannel = SmsChannel;
/** @internal */
SmsChannel.__pulumiType = 'aws:pinpoint/smsChannel:SmsChannel';
