// @pulumi/aws@7.7.0/apigatewayv2/integration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Integration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an Amazon API Gateway Version 2 integration.
 * More information can be found in the [Amazon API Gateway Developer Guide](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api.html).
 *
 * ## Example Usage
 *
 * ### Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.apigatewayv2.Integration("example", {
 *     apiId: exampleAwsApigatewayv2Api.id,
 *     integrationType: "MOCK",
 * });
 * ```
 *
 * ### Lambda Integration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.lambda.Function("example", {
 *     code: new pulumi.asset.FileArchive("example.zip"),
 *     name: "Example",
 *     role: exampleAwsIamRole.arn,
 *     handler: "index.handler",
 *     runtime: aws.lambda.Runtime.NodeJS20dX,
 * });
 * const exampleIntegration = new aws.apigatewayv2.Integration("example", {
 *     apiId: exampleAwsApigatewayv2Api.id,
 *     integrationType: "AWS_PROXY",
 *     connectionType: "INTERNET",
 *     contentHandlingStrategy: "CONVERT_TO_TEXT",
 *     description: "Lambda example",
 *     integrationMethod: "POST",
 *     integrationUri: example.invokeArn,
 *     passthroughBehavior: "WHEN_NO_MATCH",
 * });
 * ```
 *
 * ### AWS Service Integration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.apigatewayv2.Integration("example", {
 *     apiId: exampleAwsApigatewayv2Api.id,
 *     credentialsArn: exampleAwsIamRole.arn,
 *     description: "SQS example",
 *     integrationType: "AWS_PROXY",
 *     integrationSubtype: "SQS-SendMessage",
 *     requestParameters: {
 *         QueueUrl: "$request.header.queueUrl",
 *         MessageBody: "$request.body.message",
 *     },
 * });
 * ```
 *
 * ### Private Integration
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.apigatewayv2.Integration("example", {
 *     apiId: exampleAwsApigatewayv2Api.id,
 *     credentialsArn: exampleAwsIamRole.arn,
 *     description: "Example with a load balancer",
 *     integrationType: "HTTP_PROXY",
 *     integrationUri: exampleAwsLbListener.arn,
 *     integrationMethod: "ANY",
 *     connectionType: "VPC_LINK",
 *     connectionId: exampleAwsApigatewayv2VpcLink.id,
 *     tlsConfig: {
 *         serverNameToVerify: "example.com",
 *     },
 *     requestParameters: {
 *         "append:header.authforintegration": "$context.authorizer.authorizerResponse",
 *         "overwrite:path": "staticValueForIntegration",
 *     },
 *     responseParameters: [
 *         {
 *             statusCode: "403",
 *             mappings: {
 *                 "append:header.auth": "$context.authorizer.authorizerResponse",
 *             },
 *         },
 *         {
 *             statusCode: "200",
 *             mappings: {
 *                 "overwrite:statuscode": "204",
 *             },
 *         },
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_apigatewayv2_integration` using the API identifier and integration identifier. For example:
 *
 * ```sh
 * $ pulumi import aws:apigatewayv2/integration:Integration example aabbccddee/1122334
 * ```
 * -> __Note:__ The API Gateway managed integration created as part of [_quick_create_](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-basic-concept.html#apigateway-definition-quick-create) cannot be imported.
 */
class Integration extends pulumi.CustomResource {
    /**
     * Get an existing Integration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Integration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Integration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Integration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["apiId"] = state?.apiId;
            resourceInputs["connectionId"] = state?.connectionId;
            resourceInputs["connectionType"] = state?.connectionType;
            resourceInputs["contentHandlingStrategy"] = state?.contentHandlingStrategy;
            resourceInputs["credentialsArn"] = state?.credentialsArn;
            resourceInputs["description"] = state?.description;
            resourceInputs["integrationMethod"] = state?.integrationMethod;
            resourceInputs["integrationResponseSelectionExpression"] = state?.integrationResponseSelectionExpression;
            resourceInputs["integrationSubtype"] = state?.integrationSubtype;
            resourceInputs["integrationType"] = state?.integrationType;
            resourceInputs["integrationUri"] = state?.integrationUri;
            resourceInputs["passthroughBehavior"] = state?.passthroughBehavior;
            resourceInputs["payloadFormatVersion"] = state?.payloadFormatVersion;
            resourceInputs["region"] = state?.region;
            resourceInputs["requestParameters"] = state?.requestParameters;
            resourceInputs["requestTemplates"] = state?.requestTemplates;
            resourceInputs["responseParameters"] = state?.responseParameters;
            resourceInputs["templateSelectionExpression"] = state?.templateSelectionExpression;
            resourceInputs["timeoutMilliseconds"] = state?.timeoutMilliseconds;
            resourceInputs["tlsConfig"] = state?.tlsConfig;
        }
        else {
            const args = argsOrState;
            if (args?.apiId === undefined && !opts.urn) {
                throw new Error("Missing required property 'apiId'");
            }
            if (args?.integrationType === undefined && !opts.urn) {
                throw new Error("Missing required property 'integrationType'");
            }
            resourceInputs["apiId"] = args?.apiId;
            resourceInputs["connectionId"] = args?.connectionId;
            resourceInputs["connectionType"] = args?.connectionType;
            resourceInputs["contentHandlingStrategy"] = args?.contentHandlingStrategy;
            resourceInputs["credentialsArn"] = args?.credentialsArn;
            resourceInputs["description"] = args?.description;
            resourceInputs["integrationMethod"] = args?.integrationMethod;
            resourceInputs["integrationSubtype"] = args?.integrationSubtype;
            resourceInputs["integrationType"] = args?.integrationType;
            resourceInputs["integrationUri"] = args?.integrationUri;
            resourceInputs["passthroughBehavior"] = args?.passthroughBehavior;
            resourceInputs["payloadFormatVersion"] = args?.payloadFormatVersion;
            resourceInputs["region"] = args?.region;
            resourceInputs["requestParameters"] = args?.requestParameters;
            resourceInputs["requestTemplates"] = args?.requestTemplates;
            resourceInputs["responseParameters"] = args?.responseParameters;
            resourceInputs["templateSelectionExpression"] = args?.templateSelectionExpression;
            resourceInputs["timeoutMilliseconds"] = args?.timeoutMilliseconds;
            resourceInputs["tlsConfig"] = args?.tlsConfig;
            resourceInputs["integrationResponseSelectionExpression"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Integration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Integration = Integration;
/** @internal */
Integration.__pulumiType = 'aws:apigatewayv2/integration:Integration';
