// @pulumi/aws@7.7.0/elb/loadBalancerCookieStickinessPolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoadBalancerCookieStickinessPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a load balancer cookie stickiness policy, which allows an ELB to control the sticky session lifetime of the browser.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const lb = new aws.elb.LoadBalancer("lb", {
 *     name: "test-lb",
 *     availabilityZones: ["us-east-1a"],
 *     listeners: [{
 *         instancePort: 8000,
 *         instanceProtocol: "http",
 *         lbPort: 80,
 *         lbProtocol: "http",
 *     }],
 * });
 * const foo = new aws.elb.LoadBalancerCookieStickinessPolicy("foo", {
 *     name: "foo-policy",
 *     loadBalancer: lb.id,
 *     lbPort: 80,
 *     cookieExpirationPeriod: 600,
 * });
 * ```
 */
class LoadBalancerCookieStickinessPolicy extends pulumi.CustomResource {
    /**
     * Get an existing LoadBalancerCookieStickinessPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LoadBalancerCookieStickinessPolicy(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of LoadBalancerCookieStickinessPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LoadBalancerCookieStickinessPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["cookieExpirationPeriod"] = state?.cookieExpirationPeriod;
            resourceInputs["lbPort"] = state?.lbPort;
            resourceInputs["loadBalancer"] = state?.loadBalancer;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.lbPort === undefined && !opts.urn) {
                throw new Error("Missing required property 'lbPort'");
            }
            if (args?.loadBalancer === undefined && !opts.urn) {
                throw new Error("Missing required property 'loadBalancer'");
            }
            resourceInputs["cookieExpirationPeriod"] = args?.cookieExpirationPeriod;
            resourceInputs["lbPort"] = args?.lbPort;
            resourceInputs["loadBalancer"] = args?.loadBalancer;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "aws:elasticloadbalancing/loadBalancerCookieStickinessPolicy:LoadBalancerCookieStickinessPolicy" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(LoadBalancerCookieStickinessPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LoadBalancerCookieStickinessPolicy = LoadBalancerCookieStickinessPolicy;
/** @internal */
LoadBalancerCookieStickinessPolicy.__pulumiType = 'aws:elb/loadBalancerCookieStickinessPolicy:LoadBalancerCookieStickinessPolicy';
