// @pulumi/aws@7.7.0/directconnect/bgpPeer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BgpPeer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a Direct Connect BGP peer resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const peer = new aws.directconnect.BgpPeer("peer", {
 *     virtualInterfaceId: foo.id,
 *     addressFamily: "ipv6",
 *     bgpAsn: 65351,
 * });
 * ```
 */
class BgpPeer extends pulumi.CustomResource {
    /**
     * Get an existing BgpPeer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BgpPeer(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of BgpPeer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BgpPeer.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["addressFamily"] = state?.addressFamily;
            resourceInputs["amazonAddress"] = state?.amazonAddress;
            resourceInputs["awsDevice"] = state?.awsDevice;
            resourceInputs["bgpAsn"] = state?.bgpAsn;
            resourceInputs["bgpAuthKey"] = state?.bgpAuthKey;
            resourceInputs["bgpPeerId"] = state?.bgpPeerId;
            resourceInputs["bgpStatus"] = state?.bgpStatus;
            resourceInputs["customerAddress"] = state?.customerAddress;
            resourceInputs["region"] = state?.region;
            resourceInputs["virtualInterfaceId"] = state?.virtualInterfaceId;
        }
        else {
            const args = argsOrState;
            if (args?.addressFamily === undefined && !opts.urn) {
                throw new Error("Missing required property 'addressFamily'");
            }
            if (args?.bgpAsn === undefined && !opts.urn) {
                throw new Error("Missing required property 'bgpAsn'");
            }
            if (args?.virtualInterfaceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'virtualInterfaceId'");
            }
            resourceInputs["addressFamily"] = args?.addressFamily;
            resourceInputs["amazonAddress"] = args?.amazonAddress;
            resourceInputs["bgpAsn"] = args?.bgpAsn;
            resourceInputs["bgpAuthKey"] = args?.bgpAuthKey;
            resourceInputs["customerAddress"] = args?.customerAddress;
            resourceInputs["region"] = args?.region;
            resourceInputs["virtualInterfaceId"] = args?.virtualInterfaceId;
            resourceInputs["awsDevice"] = undefined /*out*/;
            resourceInputs["bgpPeerId"] = undefined /*out*/;
            resourceInputs["bgpStatus"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BgpPeer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BgpPeer = BgpPeer;
/** @internal */
BgpPeer.__pulumiType = 'aws:directconnect/bgpPeer:BgpPeer';
