// @pulumi/aws@7.7.0/directconnect/privateVirtualInterface.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateVirtualInterface = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a Direct Connect private virtual interface resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const foo = new aws.directconnect.PrivateVirtualInterface("foo", {
 *     connectionId: "dxcon-zzzzzzzz",
 *     name: "vif-foo",
 *     vlan: 4094,
 *     addressFamily: "ipv4",
 *     bgpAsn: 65352,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Direct Connect private virtual interfaces using the VIF `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:directconnect/privateVirtualInterface:PrivateVirtualInterface test dxvif-33cc44dd
 * ```
 */
class PrivateVirtualInterface extends pulumi.CustomResource {
    /**
     * Get an existing PrivateVirtualInterface resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PrivateVirtualInterface(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of PrivateVirtualInterface.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PrivateVirtualInterface.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["addressFamily"] = state?.addressFamily;
            resourceInputs["amazonAddress"] = state?.amazonAddress;
            resourceInputs["amazonSideAsn"] = state?.amazonSideAsn;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["awsDevice"] = state?.awsDevice;
            resourceInputs["bgpAsn"] = state?.bgpAsn;
            resourceInputs["bgpAuthKey"] = state?.bgpAuthKey;
            resourceInputs["connectionId"] = state?.connectionId;
            resourceInputs["customerAddress"] = state?.customerAddress;
            resourceInputs["dxGatewayId"] = state?.dxGatewayId;
            resourceInputs["jumboFrameCapable"] = state?.jumboFrameCapable;
            resourceInputs["mtu"] = state?.mtu;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["sitelinkEnabled"] = state?.sitelinkEnabled;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["vlan"] = state?.vlan;
            resourceInputs["vpnGatewayId"] = state?.vpnGatewayId;
        }
        else {
            const args = argsOrState;
            if (args?.addressFamily === undefined && !opts.urn) {
                throw new Error("Missing required property 'addressFamily'");
            }
            if (args?.bgpAsn === undefined && !opts.urn) {
                throw new Error("Missing required property 'bgpAsn'");
            }
            if (args?.connectionId === undefined && !opts.urn) {
                throw new Error("Missing required property 'connectionId'");
            }
            if (args?.vlan === undefined && !opts.urn) {
                throw new Error("Missing required property 'vlan'");
            }
            resourceInputs["addressFamily"] = args?.addressFamily;
            resourceInputs["amazonAddress"] = args?.amazonAddress;
            resourceInputs["bgpAsn"] = args?.bgpAsn;
            resourceInputs["bgpAuthKey"] = args?.bgpAuthKey;
            resourceInputs["connectionId"] = args?.connectionId;
            resourceInputs["customerAddress"] = args?.customerAddress;
            resourceInputs["dxGatewayId"] = args?.dxGatewayId;
            resourceInputs["mtu"] = args?.mtu;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["sitelinkEnabled"] = args?.sitelinkEnabled;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["vlan"] = args?.vlan;
            resourceInputs["vpnGatewayId"] = args?.vpnGatewayId;
            resourceInputs["amazonSideAsn"] = undefined /*out*/;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["awsDevice"] = undefined /*out*/;
            resourceInputs["jumboFrameCapable"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PrivateVirtualInterface.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PrivateVirtualInterface = PrivateVirtualInterface;
/** @internal */
PrivateVirtualInterface.__pulumiType = 'aws:directconnect/privateVirtualInterface:PrivateVirtualInterface';
