// @pulumi/aws@7.7.0/vpclattice/serviceNetworkVpcAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceNetworkVpcAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS VPC Lattice Service Network VPC Association.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpclattice.ServiceNetworkVpcAssociation("example", {
 *     vpcIdentifier: exampleAwsVpc.id,
 *     serviceNetworkIdentifier: exampleAwsVpclatticeServiceNetwork.id,
 *     securityGroupIds: [exampleAwsSecurityGroup.id],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import VPC Lattice Service Network VPC Association using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:vpclattice/serviceNetworkVpcAssociation:ServiceNetworkVpcAssociation example snsa-05e2474658a88f6ba
 * ```
 */
class ServiceNetworkVpcAssociation extends pulumi.CustomResource {
    /**
     * Get an existing ServiceNetworkVpcAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceNetworkVpcAssociation(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ServiceNetworkVpcAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceNetworkVpcAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["createdBy"] = state?.createdBy;
            resourceInputs["region"] = state?.region;
            resourceInputs["securityGroupIds"] = state?.securityGroupIds;
            resourceInputs["serviceNetworkIdentifier"] = state?.serviceNetworkIdentifier;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["vpcIdentifier"] = state?.vpcIdentifier;
        }
        else {
            const args = argsOrState;
            if (args?.serviceNetworkIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceNetworkIdentifier'");
            }
            if (args?.vpcIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'vpcIdentifier'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["securityGroupIds"] = args?.securityGroupIds;
            resourceInputs["serviceNetworkIdentifier"] = args?.serviceNetworkIdentifier;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["vpcIdentifier"] = args?.vpcIdentifier;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServiceNetworkVpcAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServiceNetworkVpcAssociation = ServiceNetworkVpcAssociation;
/** @internal */
ServiceNetworkVpcAssociation.__pulumiType = 'aws:vpclattice/serviceNetworkVpcAssociation:ServiceNetworkVpcAssociation';
