// @pulumi/aws@7.7.0/vpclattice/serviceNetworkServiceAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceNetworkServiceAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS VPC Lattice Service Network Service Association.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpclattice.ServiceNetworkServiceAssociation("example", {
 *     serviceIdentifier: exampleAwsVpclatticeService.id,
 *     serviceNetworkIdentifier: exampleAwsVpclatticeServiceNetwork.id,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import VPC Lattice Service Network Service Association using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:vpclattice/serviceNetworkServiceAssociation:ServiceNetworkServiceAssociation example snsa-05e2474658a88f6ba
 * ```
 */
class ServiceNetworkServiceAssociation extends pulumi.CustomResource {
    /**
     * Get an existing ServiceNetworkServiceAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ServiceNetworkServiceAssociation(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ServiceNetworkServiceAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ServiceNetworkServiceAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["createdBy"] = state?.createdBy;
            resourceInputs["customDomainName"] = state?.customDomainName;
            resourceInputs["dnsEntries"] = state?.dnsEntries;
            resourceInputs["region"] = state?.region;
            resourceInputs["serviceIdentifier"] = state?.serviceIdentifier;
            resourceInputs["serviceNetworkIdentifier"] = state?.serviceNetworkIdentifier;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.serviceIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceIdentifier'");
            }
            if (args?.serviceNetworkIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceNetworkIdentifier'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["serviceIdentifier"] = args?.serviceIdentifier;
            resourceInputs["serviceNetworkIdentifier"] = args?.serviceNetworkIdentifier;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["customDomainName"] = undefined /*out*/;
            resourceInputs["dnsEntries"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ServiceNetworkServiceAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ServiceNetworkServiceAssociation = ServiceNetworkServiceAssociation;
/** @internal */
ServiceNetworkServiceAssociation.__pulumiType = 'aws:vpclattice/serviceNetworkServiceAssociation:ServiceNetworkServiceAssociation';
