// @pulumi/aws@7.7.0/iot/domainConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DomainConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Creates and manages an AWS IoT domain configuration.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const iot = new aws.iot.DomainConfiguration("iot", {
 *     name: "iot-",
 *     domainName: "iot.example.com",
 *     serviceType: "DATA",
 *     serverCertificateArns: [cert.arn],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import domain configurations using the name. For example:
 *
 * ```sh
 * $ pulumi import aws:iot/domainConfiguration:DomainConfiguration example example
 * ```
 */
class DomainConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing DomainConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DomainConfiguration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DomainConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DomainConfiguration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["applicationProtocol"] = state?.applicationProtocol;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["authenticationType"] = state?.authenticationType;
            resourceInputs["authorizerConfig"] = state?.authorizerConfig;
            resourceInputs["domainName"] = state?.domainName;
            resourceInputs["domainType"] = state?.domainType;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["serverCertificateArns"] = state?.serverCertificateArns;
            resourceInputs["serviceType"] = state?.serviceType;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["tlsConfig"] = state?.tlsConfig;
            resourceInputs["validationCertificateArn"] = state?.validationCertificateArn;
        }
        else {
            const args = argsOrState;
            resourceInputs["applicationProtocol"] = args?.applicationProtocol;
            resourceInputs["authenticationType"] = args?.authenticationType;
            resourceInputs["authorizerConfig"] = args?.authorizerConfig;
            resourceInputs["domainName"] = args?.domainName;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["serverCertificateArns"] = args?.serverCertificateArns;
            resourceInputs["serviceType"] = args?.serviceType;
            resourceInputs["status"] = args?.status;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["tlsConfig"] = args?.tlsConfig;
            resourceInputs["validationCertificateArn"] = args?.validationCertificateArn;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["domainType"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DomainConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DomainConfiguration = DomainConfiguration;
/** @internal */
DomainConfiguration.__pulumiType = 'aws:iot/domainConfiguration:DomainConfiguration';
