// @pulumi/aws@7.7.0/iot/thingPrincipalAttachment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThingPrincipalAttachment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Attaches Principal to AWS IoT Thing.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 * import * as std from "@pulumi/std";
 *
 * const example = new aws.iot.Thing("example", {name: "example"});
 * const cert = new aws.iot.Certificate("cert", {
 *     csr: std.file({
 *         input: "csr.pem",
 *     }).then(invoke => invoke.result),
 *     active: true,
 * });
 * const att = new aws.iot.ThingPrincipalAttachment("att", {
 *     principal: cert.arn,
 *     thing: example.name,
 * });
 * ```
 */
class ThingPrincipalAttachment extends pulumi.CustomResource {
    /**
     * Get an existing ThingPrincipalAttachment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ThingPrincipalAttachment(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ThingPrincipalAttachment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ThingPrincipalAttachment.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["principal"] = state?.principal;
            resourceInputs["region"] = state?.region;
            resourceInputs["thing"] = state?.thing;
            resourceInputs["thingPrincipalType"] = state?.thingPrincipalType;
        }
        else {
            const args = argsOrState;
            if (args?.principal === undefined && !opts.urn) {
                throw new Error("Missing required property 'principal'");
            }
            if (args?.thing === undefined && !opts.urn) {
                throw new Error("Missing required property 'thing'");
            }
            resourceInputs["principal"] = args?.principal;
            resourceInputs["region"] = args?.region;
            resourceInputs["thing"] = args?.thing;
            resourceInputs["thingPrincipalType"] = args?.thingPrincipalType;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ThingPrincipalAttachment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ThingPrincipalAttachment = ThingPrincipalAttachment;
/** @internal */
ThingPrincipalAttachment.__pulumiType = 'aws:iot/thingPrincipalAttachment:ThingPrincipalAttachment';
