// @pulumi/aws@7.7.0/docdb/eventSubscription.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a DocumentDB event subscription resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.docdb.Cluster("example", {
 *     clusterIdentifier: "example",
 *     availabilityZones: [
 *         available.names[0],
 *         available.names[1],
 *         available.names[2],
 *     ],
 *     masterUsername: "foo",
 *     masterPassword: "mustbeeightcharaters",
 *     skipFinalSnapshot: true,
 * });
 * const exampleTopic = new aws.sns.Topic("example", {name: "example-events"});
 * const exampleEventSubscription = new aws.docdb.EventSubscription("example", {
 *     name: "example",
 *     enabled: true,
 *     eventCategories: [
 *         "creation",
 *         "failure",
 *     ],
 *     sourceType: "db-cluster",
 *     sourceIds: [example.id],
 *     snsTopicArn: exampleTopic.arn,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import DocumentDB Event Subscriptions using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:docdb/eventSubscription:EventSubscription example event-sub
 * ```
 */
class EventSubscription extends pulumi.CustomResource {
    /**
     * Get an existing EventSubscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EventSubscription(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EventSubscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EventSubscription.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["customerAwsId"] = state?.customerAwsId;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["eventCategories"] = state?.eventCategories;
            resourceInputs["name"] = state?.name;
            resourceInputs["namePrefix"] = state?.namePrefix;
            resourceInputs["region"] = state?.region;
            resourceInputs["snsTopicArn"] = state?.snsTopicArn;
            resourceInputs["sourceIds"] = state?.sourceIds;
            resourceInputs["sourceType"] = state?.sourceType;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.snsTopicArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'snsTopicArn'");
            }
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["eventCategories"] = args?.eventCategories;
            resourceInputs["name"] = args?.name;
            resourceInputs["namePrefix"] = args?.namePrefix;
            resourceInputs["region"] = args?.region;
            resourceInputs["snsTopicArn"] = args?.snsTopicArn;
            resourceInputs["sourceIds"] = args?.sourceIds;
            resourceInputs["sourceType"] = args?.sourceType;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["customerAwsId"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EventSubscription.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EventSubscription = EventSubscription;
/** @internal */
EventSubscription.__pulumiType = 'aws:docdb/eventSubscription:EventSubscription';
