// @pulumi/aws@7.7.0/verifiedpermissions/schema.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Schema = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.verifiedpermissions.Schema("example", {
 *     policyStoreId: exampleAwsVerifiedpermissionsPolicyStore.policyStoreId,
 *     definition: {
 *         value: JSON.stringify({
 *             Namespace: {
 *                 entityTypes: {},
 *                 actions: {},
 *             },
 *         }),
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Verified Permissions Policy Store Schema using the `policy_store_id`. For example:
 *
 * console
 *
 *  % pulumi import aws_verifiedpermissions_schema.example DxQg2j8xvXJQ1tQCYNWj9T
 */
class Schema extends pulumi.CustomResource {
    /**
     * Get an existing Schema resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Schema(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Schema.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Schema.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["definition"] = state?.definition;
            resourceInputs["namespaces"] = state?.namespaces;
            resourceInputs["policyStoreId"] = state?.policyStoreId;
            resourceInputs["region"] = state?.region;
        }
        else {
            const args = argsOrState;
            if (args?.policyStoreId === undefined && !opts.urn) {
                throw new Error("Missing required property 'policyStoreId'");
            }
            resourceInputs["definition"] = args?.definition;
            resourceInputs["policyStoreId"] = args?.policyStoreId;
            resourceInputs["region"] = args?.region;
            resourceInputs["namespaces"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Schema.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Schema = Schema;
/** @internal */
Schema.__pulumiType = 'aws:verifiedpermissions/schema:Schema';
