// @pulumi/aws@7.7.0/appstream/fleet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Fleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an AppStream fleet.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const testFleet = new aws.appstream.Fleet("test_fleet", {
 *     name: "test-fleet",
 *     computeCapacity: {
 *         desiredInstances: 1,
 *     },
 *     description: "test fleet",
 *     idleDisconnectTimeoutInSeconds: 60,
 *     displayName: "test-fleet",
 *     enableDefaultInternetAccess: false,
 *     fleetType: "ON_DEMAND",
 *     imageName: "Amazon-AppStream2-Sample-Image-03-11-2023",
 *     instanceType: "stream.standard.large",
 *     maxUserDurationInSeconds: 600,
 *     vpcConfig: {
 *         subnetIds: ["subnet-06e9b13400c225127"],
 *     },
 *     tags: {
 *         TagName: "tag-value",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_appstream_fleet` using the id. For example:
 *
 * ```sh
 * $ pulumi import aws:appstream/fleet:Fleet example fleetNameExample
 * ```
 */
class Fleet extends pulumi.CustomResource {
    /**
     * Get an existing Fleet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Fleet(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Fleet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Fleet.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["computeCapacity"] = state?.computeCapacity;
            resourceInputs["createdTime"] = state?.createdTime;
            resourceInputs["description"] = state?.description;
            resourceInputs["disconnectTimeoutInSeconds"] = state?.disconnectTimeoutInSeconds;
            resourceInputs["displayName"] = state?.displayName;
            resourceInputs["domainJoinInfo"] = state?.domainJoinInfo;
            resourceInputs["enableDefaultInternetAccess"] = state?.enableDefaultInternetAccess;
            resourceInputs["fleetType"] = state?.fleetType;
            resourceInputs["iamRoleArn"] = state?.iamRoleArn;
            resourceInputs["idleDisconnectTimeoutInSeconds"] = state?.idleDisconnectTimeoutInSeconds;
            resourceInputs["imageArn"] = state?.imageArn;
            resourceInputs["imageName"] = state?.imageName;
            resourceInputs["instanceType"] = state?.instanceType;
            resourceInputs["maxSessionsPerInstance"] = state?.maxSessionsPerInstance;
            resourceInputs["maxUserDurationInSeconds"] = state?.maxUserDurationInSeconds;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["state"] = state?.state;
            resourceInputs["streamView"] = state?.streamView;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["vpcConfig"] = state?.vpcConfig;
        }
        else {
            const args = argsOrState;
            if (args?.computeCapacity === undefined && !opts.urn) {
                throw new Error("Missing required property 'computeCapacity'");
            }
            if (args?.instanceType === undefined && !opts.urn) {
                throw new Error("Missing required property 'instanceType'");
            }
            resourceInputs["computeCapacity"] = args?.computeCapacity;
            resourceInputs["description"] = args?.description;
            resourceInputs["disconnectTimeoutInSeconds"] = args?.disconnectTimeoutInSeconds;
            resourceInputs["displayName"] = args?.displayName;
            resourceInputs["domainJoinInfo"] = args?.domainJoinInfo;
            resourceInputs["enableDefaultInternetAccess"] = args?.enableDefaultInternetAccess;
            resourceInputs["fleetType"] = args?.fleetType;
            resourceInputs["iamRoleArn"] = args?.iamRoleArn;
            resourceInputs["idleDisconnectTimeoutInSeconds"] = args?.idleDisconnectTimeoutInSeconds;
            resourceInputs["imageArn"] = args?.imageArn;
            resourceInputs["imageName"] = args?.imageName;
            resourceInputs["instanceType"] = args?.instanceType;
            resourceInputs["maxSessionsPerInstance"] = args?.maxSessionsPerInstance;
            resourceInputs["maxUserDurationInSeconds"] = args?.maxUserDurationInSeconds;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["streamView"] = args?.streamView;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["vpcConfig"] = args?.vpcConfig;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Fleet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Fleet = Fleet;
/** @internal */
Fleet.__pulumiType = 'aws:appstream/fleet:Fleet';
