// @pulumi/aws@7.7.0/cognito/riskConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RiskConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a Cognito Risk Configuration resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cognito.RiskConfiguration("example", {
 *     userPoolId: exampleAwsCognitoUserPool.id,
 *     riskExceptionConfiguration: {
 *         blockedIpRangeLists: ["10.10.10.10/32"],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Import using the user pool ID and Client ID separated by a `:`:
 *
 * __Using `pulumi import` to import__ Cognito Risk Configurations using the user pool ID or the user pool ID and Client Id separated by a `:`. For example:
 *
 * Import using the user pool ID:
 *
 * ```sh
 * $ pulumi import aws:cognito/riskConfiguration:RiskConfiguration main example
 * ```
 * Import using the user pool ID and Client ID separated by a `:`:
 *
 * ```sh
 * $ pulumi import aws:cognito/riskConfiguration:RiskConfiguration main example:example
 * ```
 */
class RiskConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing RiskConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RiskConfiguration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RiskConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RiskConfiguration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountTakeoverRiskConfiguration"] = state?.accountTakeoverRiskConfiguration;
            resourceInputs["clientId"] = state?.clientId;
            resourceInputs["compromisedCredentialsRiskConfiguration"] = state?.compromisedCredentialsRiskConfiguration;
            resourceInputs["region"] = state?.region;
            resourceInputs["riskExceptionConfiguration"] = state?.riskExceptionConfiguration;
            resourceInputs["userPoolId"] = state?.userPoolId;
        }
        else {
            const args = argsOrState;
            if (args?.userPoolId === undefined && !opts.urn) {
                throw new Error("Missing required property 'userPoolId'");
            }
            resourceInputs["accountTakeoverRiskConfiguration"] = args?.accountTakeoverRiskConfiguration;
            resourceInputs["clientId"] = args?.clientId;
            resourceInputs["compromisedCredentialsRiskConfiguration"] = args?.compromisedCredentialsRiskConfiguration;
            resourceInputs["region"] = args?.region;
            resourceInputs["riskExceptionConfiguration"] = args?.riskExceptionConfiguration;
            resourceInputs["userPoolId"] = args?.userPoolId;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RiskConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RiskConfiguration = RiskConfiguration;
/** @internal */
RiskConfiguration.__pulumiType = 'aws:cognito/riskConfiguration:RiskConfiguration';
