// @pulumi/aws@7.7.0/datazone/userProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS DataZone User Profile.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.datazone.UserProfile("example", {
 *     userIdentifier: exampleAwsIamUser.arn,
 *     domainIdentifier: exampleAwsDatazoneDomain.id,
 *     userType: "IAM_USER",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import DataZone User Profile using the `user_identifier,domain_identifier,type`. For example:
 *
 * ```sh
 * $ pulumi import aws:datazone/userProfile:UserProfile example arn:aws:iam::123456789012:user/example,dzd_54nakfrg9k6suo,IAM
 * ```
 */
class UserProfile extends pulumi.CustomResource {
    /**
     * Get an existing UserProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new UserProfile(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of UserProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserProfile.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["details"] = state?.details;
            resourceInputs["domainIdentifier"] = state?.domainIdentifier;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["type"] = state?.type;
            resourceInputs["userIdentifier"] = state?.userIdentifier;
            resourceInputs["userType"] = state?.userType;
        }
        else {
            const args = argsOrState;
            if (args?.domainIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'domainIdentifier'");
            }
            if (args?.userIdentifier === undefined && !opts.urn) {
                throw new Error("Missing required property 'userIdentifier'");
            }
            resourceInputs["domainIdentifier"] = args?.domainIdentifier;
            resourceInputs["region"] = args?.region;
            resourceInputs["status"] = args?.status;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["userIdentifier"] = args?.userIdentifier;
            resourceInputs["userType"] = args?.userType;
            resourceInputs["details"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(UserProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.UserProfile = UserProfile;
/** @internal */
UserProfile.__pulumiType = 'aws:datazone/userProfile:UserProfile';
