// @pulumi/aws@7.7.0/inspector2/memberAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MemberAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for associating accounts to existing Inspector instances.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.inspector2.MemberAssociation("example", {accountId: "123456789012"});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Amazon Inspector Member Association using the `account_id`. For example:
 *
 * ```sh
 * $ pulumi import aws:inspector2/memberAssociation:MemberAssociation example 123456789012
 * ```
 */
class MemberAssociation extends pulumi.CustomResource {
    /**
     * Get an existing MemberAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MemberAssociation(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of MemberAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MemberAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["delegatedAdminAccountId"] = state?.delegatedAdminAccountId;
            resourceInputs["region"] = state?.region;
            resourceInputs["relationshipStatus"] = state?.relationshipStatus;
            resourceInputs["updatedAt"] = state?.updatedAt;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["region"] = args?.region;
            resourceInputs["delegatedAdminAccountId"] = undefined /*out*/;
            resourceInputs["relationshipStatus"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MemberAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MemberAssociation = MemberAssociation;
/** @internal */
MemberAssociation.__pulumiType = 'aws:inspector2/memberAssociation:MemberAssociation';
