// @pulumi/aws@7.7.0/networkfirewall/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcEndpointAssociation = exports.TlsInspectionConfiguration = exports.RuleGroup = exports.ResourcePolicy = exports.LoggingConfiguration = exports.getResourcePolicyOutput = exports.getResourcePolicy = exports.getFirewallPolicyOutput = exports.getFirewallPolicy = exports.getFirewallOutput = exports.getFirewall = exports.FirewallTransitGatewayAttachmentAccepter = exports.FirewallPolicy = exports.Firewall = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Firewall = null;
utilities.lazyLoad(exports, ["Firewall"], () => require("./firewall"));
exports.FirewallPolicy = null;
utilities.lazyLoad(exports, ["FirewallPolicy"], () => require("./firewallPolicy"));
exports.FirewallTransitGatewayAttachmentAccepter = null;
utilities.lazyLoad(exports, ["FirewallTransitGatewayAttachmentAccepter"], () => require("./firewallTransitGatewayAttachmentAccepter"));
exports.getFirewall = null;
exports.getFirewallOutput = null;
utilities.lazyLoad(exports, ["getFirewall", "getFirewallOutput"], () => require("./getFirewall"));
exports.getFirewallPolicy = null;
exports.getFirewallPolicyOutput = null;
utilities.lazyLoad(exports, ["getFirewallPolicy", "getFirewallPolicyOutput"], () => require("./getFirewallPolicy"));
exports.getResourcePolicy = null;
exports.getResourcePolicyOutput = null;
utilities.lazyLoad(exports, ["getResourcePolicy", "getResourcePolicyOutput"], () => require("./getResourcePolicy"));
exports.LoggingConfiguration = null;
utilities.lazyLoad(exports, ["LoggingConfiguration"], () => require("./loggingConfiguration"));
exports.ResourcePolicy = null;
utilities.lazyLoad(exports, ["ResourcePolicy"], () => require("./resourcePolicy"));
exports.RuleGroup = null;
utilities.lazyLoad(exports, ["RuleGroup"], () => require("./ruleGroup"));
exports.TlsInspectionConfiguration = null;
utilities.lazyLoad(exports, ["TlsInspectionConfiguration"], () => require("./tlsInspectionConfiguration"));
exports.VpcEndpointAssociation = null;
utilities.lazyLoad(exports, ["VpcEndpointAssociation"], () => require("./vpcEndpointAssociation"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:networkfirewall/firewall:Firewall":
                return new exports.Firewall(name, undefined, { urn });
            case "aws:networkfirewall/firewallPolicy:FirewallPolicy":
                return new exports.FirewallPolicy(name, undefined, { urn });
            case "aws:networkfirewall/firewallTransitGatewayAttachmentAccepter:FirewallTransitGatewayAttachmentAccepter":
                return new exports.FirewallTransitGatewayAttachmentAccepter(name, undefined, { urn });
            case "aws:networkfirewall/loggingConfiguration:LoggingConfiguration":
                return new exports.LoggingConfiguration(name, undefined, { urn });
            case "aws:networkfirewall/resourcePolicy:ResourcePolicy":
                return new exports.ResourcePolicy(name, undefined, { urn });
            case "aws:networkfirewall/ruleGroup:RuleGroup":
                return new exports.RuleGroup(name, undefined, { urn });
            case "aws:networkfirewall/tlsInspectionConfiguration:TlsInspectionConfiguration":
                return new exports.TlsInspectionConfiguration(name, undefined, { urn });
            case "aws:networkfirewall/vpcEndpointAssociation:VpcEndpointAssociation":
                return new exports.VpcEndpointAssociation(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "networkfirewall/firewall", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/firewallPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/firewallTransitGatewayAttachmentAccepter", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/loggingConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/resourcePolicy", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/ruleGroup", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/tlsInspectionConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "networkfirewall/vpcEndpointAssociation", _module);
