// @pulumi/aws@7.7.0/vpc/endpointServicePrivateDnsVerification.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointServicePrivateDnsVerification = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS VPC (Virtual Private Cloud) Endpoint Service Private DNS Verification.
 * This resource begins the verification process by calling the [`StartVpcEndpointServicePrivateDnsVerification`](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_StartVpcEndpointServicePrivateDnsVerification.html) API.
 * The service provider should add a record to the DNS server _before_ creating this resource.
 *
 * For additional details, refer to the AWS documentation on [managing VPC endpoint service DNS names](https://docs.aws.amazon.com/vpc/latest/privatelink/manage-dns-names.html).
 *
 * > Destruction of this resource will not stop the verification process, only remove the resource from state.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.vpc.EndpointServicePrivateDnsVerification("example", {serviceId: exampleAwsVpcEndpointService.id});
 * ```
 *
 * ## Import
 *
 * You cannot import this resource.
 */
class EndpointServicePrivateDnsVerification extends pulumi.CustomResource {
    /**
     * Get an existing EndpointServicePrivateDnsVerification resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EndpointServicePrivateDnsVerification(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EndpointServicePrivateDnsVerification.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointServicePrivateDnsVerification.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["region"] = state?.region;
            resourceInputs["serviceId"] = state?.serviceId;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["waitForVerification"] = state?.waitForVerification;
        }
        else {
            const args = argsOrState;
            if (args?.serviceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'serviceId'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["serviceId"] = args?.serviceId;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["waitForVerification"] = args?.waitForVerification;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EndpointServicePrivateDnsVerification.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EndpointServicePrivateDnsVerification = EndpointServicePrivateDnsVerification;
/** @internal */
EndpointServicePrivateDnsVerification.__pulumiType = 'aws:vpc/endpointServicePrivateDnsVerification:EndpointServicePrivateDnsVerification';
