// @pulumi/aws@7.7.0/vpc/routeServerEndpoint.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteServerEndpoint = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a resource for managing a VPC (Virtual Private Cloud) Route Server Endpoint.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.vpc.RouteServerEndpoint("test", {
 *     routeServerId: example.routeServerId,
 *     subnetId: main.id,
 *     tags: {
 *         Name: "Endpoint A",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import VPC (Virtual Private Cloud) Route Server Endpoint using the `route_server_endpoint_id`. For example:
 *
 * ```sh
 * $ pulumi import aws:vpc/routeServerEndpoint:RouteServerEndpoint example rse-12345678
 * ```
 */
class RouteServerEndpoint extends pulumi.CustomResource {
    /**
     * Get an existing RouteServerEndpoint resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RouteServerEndpoint(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RouteServerEndpoint.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RouteServerEndpoint.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["eniAddress"] = state?.eniAddress;
            resourceInputs["eniId"] = state?.eniId;
            resourceInputs["region"] = state?.region;
            resourceInputs["routeServerEndpointId"] = state?.routeServerEndpointId;
            resourceInputs["routeServerId"] = state?.routeServerId;
            resourceInputs["subnetId"] = state?.subnetId;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["timeouts"] = state?.timeouts;
            resourceInputs["vpcId"] = state?.vpcId;
        }
        else {
            const args = argsOrState;
            if (args?.routeServerId === undefined && !opts.urn) {
                throw new Error("Missing required property 'routeServerId'");
            }
            if (args?.subnetId === undefined && !opts.urn) {
                throw new Error("Missing required property 'subnetId'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["routeServerId"] = args?.routeServerId;
            resourceInputs["subnetId"] = args?.subnetId;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["timeouts"] = args?.timeouts;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["eniAddress"] = undefined /*out*/;
            resourceInputs["eniId"] = undefined /*out*/;
            resourceInputs["routeServerEndpointId"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["vpcId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RouteServerEndpoint.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RouteServerEndpoint = RouteServerEndpoint;
/** @internal */
RouteServerEndpoint.__pulumiType = 'aws:vpc/routeServerEndpoint:RouteServerEndpoint';
