// @pulumi/aws@7.7.0/chime/sdkvoiceSipMediaApplication.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SdkvoiceSipMediaApplication = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A ChimeSDKVoice SIP Media Application is a managed object that passes values from a SIP rule to a target AWS Lambda function.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.chime.SdkvoiceSipMediaApplication("example", {
 *     awsRegion: "us-east-1",
 *     name: "example-sip-media-application",
 *     endpoints: {
 *         lambdaArn: test.arn,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import a ChimeSDKVoice SIP Media Application using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:chime/sdkvoiceSipMediaApplication:SdkvoiceSipMediaApplication example abcdef123456
 * ```
 */
class SdkvoiceSipMediaApplication extends pulumi.CustomResource {
    /**
     * Get an existing SdkvoiceSipMediaApplication resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SdkvoiceSipMediaApplication(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of SdkvoiceSipMediaApplication.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SdkvoiceSipMediaApplication.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["awsRegion"] = state?.awsRegion;
            resourceInputs["endpoints"] = state?.endpoints;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.awsRegion === undefined && !opts.urn) {
                throw new Error("Missing required property 'awsRegion'");
            }
            if (args?.endpoints === undefined && !opts.urn) {
                throw new Error("Missing required property 'endpoints'");
            }
            resourceInputs["awsRegion"] = args?.awsRegion;
            resourceInputs["endpoints"] = args?.endpoints;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SdkvoiceSipMediaApplication.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SdkvoiceSipMediaApplication = SdkvoiceSipMediaApplication;
/** @internal */
SdkvoiceSipMediaApplication.__pulumiType = 'aws:chime/sdkvoiceSipMediaApplication:SdkvoiceSipMediaApplication';
