// @pulumi/aws@7.7.0/chime/voiceConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoiceConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Enables you to connect your phone system to the telephone network at a substantial cost savings by using SIP trunking.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const test = new aws.chime.VoiceConnector("test", {
 *     name: "connector-test-1",
 *     requireEncryption: true,
 *     awsRegion: "us-east-1",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Configuration Recorder using the name. For example:
 *
 * ```sh
 * $ pulumi import aws:chime/voiceConnector:VoiceConnector test example
 * ```
 */
class VoiceConnector extends pulumi.CustomResource {
    /**
     * Get an existing VoiceConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new VoiceConnector(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of VoiceConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === VoiceConnector.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["awsRegion"] = state?.awsRegion;
            resourceInputs["name"] = state?.name;
            resourceInputs["outboundHostName"] = state?.outboundHostName;
            resourceInputs["region"] = state?.region;
            resourceInputs["requireEncryption"] = state?.requireEncryption;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.requireEncryption === undefined && !opts.urn) {
                throw new Error("Missing required property 'requireEncryption'");
            }
            resourceInputs["awsRegion"] = args?.awsRegion;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["requireEncryption"] = args?.requireEncryption;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["outboundHostName"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(VoiceConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.VoiceConnector = VoiceConnector;
/** @internal */
VoiceConnector.__pulumiType = 'aws:chime/voiceConnector:VoiceConnector';
