// @pulumi/aws@7.7.0/finspace/kxScalingGroup.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.KxScalingGroup = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS FinSpace Kx Scaling Group.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.finspace.KxScalingGroup("example", {
 *     name: "my-tf-kx-scalinggroup",
 *     environmentId: exampleAwsFinspaceKxEnvironment.id,
 *     availabilityZoneId: "use1-az2",
 *     hostType: "kx.sg.4xlarge",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import an AWS FinSpace Kx Scaling Group using the `id` (environment ID and scaling group name, comma-delimited). For example:
 *
 * ```sh
 * $ pulumi import aws:finspace/kxScalingGroup:KxScalingGroup example n3ceo7wqxoxcti5tujqwzs,my-tf-kx-scalinggroup
 * ```
 */
class KxScalingGroup extends pulumi.CustomResource {
    /**
     * Get an existing KxScalingGroup resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new KxScalingGroup(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of KxScalingGroup.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === KxScalingGroup.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["availabilityZoneId"] = state?.availabilityZoneId;
            resourceInputs["clusters"] = state?.clusters;
            resourceInputs["createdTimestamp"] = state?.createdTimestamp;
            resourceInputs["environmentId"] = state?.environmentId;
            resourceInputs["hostType"] = state?.hostType;
            resourceInputs["lastModifiedTimestamp"] = state?.lastModifiedTimestamp;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["statusReason"] = state?.statusReason;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.availabilityZoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'availabilityZoneId'");
            }
            if (args?.environmentId === undefined && !opts.urn) {
                throw new Error("Missing required property 'environmentId'");
            }
            if (args?.hostType === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostType'");
            }
            resourceInputs["availabilityZoneId"] = args?.availabilityZoneId;
            resourceInputs["environmentId"] = args?.environmentId;
            resourceInputs["hostType"] = args?.hostType;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["clusters"] = undefined /*out*/;
            resourceInputs["createdTimestamp"] = undefined /*out*/;
            resourceInputs["lastModifiedTimestamp"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["statusReason"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(KxScalingGroup.__pulumiType, name, resourceInputs, opts);
    }
}
exports.KxScalingGroup = KxScalingGroup;
/** @internal */
KxScalingGroup.__pulumiType = 'aws:finspace/kxScalingGroup:KxScalingGroup';
