// @pulumi/aws@7.7.0/transfer/workflow.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Workflow = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a AWS Transfer Workflow resource.
 *
 * ## Example Usage
 *
 * ### Basic single step example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.transfer.Workflow("example", {steps: [{
 *     deleteStepDetails: {
 *         name: "example",
 *         sourceFileLocation: "${original.file}",
 *     },
 *     type: "DELETE",
 * }]});
 * ```
 *
 * ### Multistep example
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.transfer.Workflow("example", {steps: [
 *     {
 *         customStepDetails: {
 *             name: "example",
 *             sourceFileLocation: "${original.file}",
 *             target: exampleAwsLambdaFunction.arn,
 *             timeoutSeconds: 60,
 *         },
 *         type: "CUSTOM",
 *     },
 *     {
 *         tagStepDetails: {
 *             name: "example",
 *             sourceFileLocation: "${original.file}",
 *             tags: [{
 *                 key: "Name",
 *                 value: "Hello World",
 *             }],
 *         },
 *         type: "TAG",
 *     },
 * ]});
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Transfer Workflows using the `worflow_id`. For example:
 *
 * ```sh
 * $ pulumi import aws:transfer/workflow:Workflow example example
 * ```
 */
class Workflow extends pulumi.CustomResource {
    /**
     * Get an existing Workflow resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Workflow(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Workflow.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Workflow.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["description"] = state?.description;
            resourceInputs["onExceptionSteps"] = state?.onExceptionSteps;
            resourceInputs["region"] = state?.region;
            resourceInputs["steps"] = state?.steps;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.steps === undefined && !opts.urn) {
                throw new Error("Missing required property 'steps'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["onExceptionSteps"] = args?.onExceptionSteps;
            resourceInputs["region"] = args?.region;
            resourceInputs["steps"] = args?.steps;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Workflow.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Workflow = Workflow;
/** @internal */
Workflow.__pulumiType = 'aws:transfer/workflow:Workflow';
