// @pulumi/aws@7.7.0/s3tables/tablePolicy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TablePolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an Amazon S3 Tables Table Policy.
 *
 * ## Example Usage
 *
 * ## Import
 *
 * Using `pulumi import`, import S3 Tables Table Policy using the `table_bucket_arn`, the value of `namespace`, and the value of `name`, separated by a semicolon (`;`). For example:
 *
 * ```sh
 * $ pulumi import aws:s3tables/tablePolicy:TablePolicy example 'arn:aws:s3tables:us-west-2:123456789012:bucket/example-bucket;example-namespace;example-table'
 * ```
 */
class TablePolicy extends pulumi.CustomResource {
    /**
     * Get an existing TablePolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TablePolicy(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of TablePolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TablePolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["name"] = state?.name;
            resourceInputs["namespace"] = state?.namespace;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourcePolicy"] = state?.resourcePolicy;
            resourceInputs["tableBucketArn"] = state?.tableBucketArn;
        }
        else {
            const args = argsOrState;
            if (args?.namespace === undefined && !opts.urn) {
                throw new Error("Missing required property 'namespace'");
            }
            if (args?.resourcePolicy === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourcePolicy'");
            }
            if (args?.tableBucketArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'tableBucketArn'");
            }
            resourceInputs["name"] = args?.name;
            resourceInputs["namespace"] = args?.namespace;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourcePolicy"] = args?.resourcePolicy;
            resourceInputs["tableBucketArn"] = args?.tableBucketArn;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TablePolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TablePolicy = TablePolicy;
/** @internal */
TablePolicy.__pulumiType = 'aws:s3tables/tablePolicy:TablePolicy';
