// @pulumi/aws@7.7.0/cloudwatch/logDeliverySource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.LogDeliverySource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing an AWS CloudWatch Logs Delivery Source.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.cloudwatch.LogDeliverySource("example", {
 *     name: "example",
 *     logType: "APPLICATION_LOGS",
 *     resourceArn: exampleAwsBedrockagentKnowledgeBase.arn,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import CloudWatch Logs Delivery Source using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:cloudwatch/logDeliverySource:LogDeliverySource example example
 * ```
 */
class LogDeliverySource extends pulumi.CustomResource {
    /**
     * Get an existing LogDeliverySource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new LogDeliverySource(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of LogDeliverySource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === LogDeliverySource.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["logType"] = state?.logType;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["resourceArn"] = state?.resourceArn;
            resourceInputs["service"] = state?.service;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.logType === undefined && !opts.urn) {
                throw new Error("Missing required property 'logType'");
            }
            if (args?.resourceArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'resourceArn'");
            }
            resourceInputs["logType"] = args?.logType;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["resourceArn"] = args?.resourceArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["service"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(LogDeliverySource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.LogDeliverySource = LogDeliverySource;
/** @internal */
LogDeliverySource.__pulumiType = 'aws:cloudwatch/logDeliverySource:LogDeliverySource';
