// @pulumi/aws@7.7.0/verifiedaccess/trustProvider.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TrustProvider = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing a Verified Access Trust Provider.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.verifiedaccess.TrustProvider("example", {
 *     policyReferenceName: "example",
 *     trustProviderType: "user",
 *     userTrustProviderType: "iam-identity-center",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Transfer Workflows using the  `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:verifiedaccess/trustProvider:TrustProvider example vatp-8012925589
 * ```
 */
class TrustProvider extends pulumi.CustomResource {
    /**
     * Get an existing TrustProvider resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TrustProvider(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of TrustProvider.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TrustProvider.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["description"] = state?.description;
            resourceInputs["deviceOptions"] = state?.deviceOptions;
            resourceInputs["deviceTrustProviderType"] = state?.deviceTrustProviderType;
            resourceInputs["nativeApplicationOidcOptions"] = state?.nativeApplicationOidcOptions;
            resourceInputs["oidcOptions"] = state?.oidcOptions;
            resourceInputs["policyReferenceName"] = state?.policyReferenceName;
            resourceInputs["region"] = state?.region;
            resourceInputs["sseSpecification"] = state?.sseSpecification;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["trustProviderType"] = state?.trustProviderType;
            resourceInputs["userTrustProviderType"] = state?.userTrustProviderType;
        }
        else {
            const args = argsOrState;
            if (args?.policyReferenceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'policyReferenceName'");
            }
            if (args?.trustProviderType === undefined && !opts.urn) {
                throw new Error("Missing required property 'trustProviderType'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["deviceOptions"] = args?.deviceOptions;
            resourceInputs["deviceTrustProviderType"] = args?.deviceTrustProviderType;
            resourceInputs["nativeApplicationOidcOptions"] = args?.nativeApplicationOidcOptions;
            resourceInputs["oidcOptions"] = args?.oidcOptions;
            resourceInputs["policyReferenceName"] = args?.policyReferenceName;
            resourceInputs["region"] = args?.region;
            resourceInputs["sseSpecification"] = args?.sseSpecification;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["trustProviderType"] = args?.trustProviderType;
            resourceInputs["userTrustProviderType"] = args?.userTrustProviderType;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TrustProvider.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TrustProvider = TrustProvider;
/** @internal */
TrustProvider.__pulumiType = 'aws:verifiedaccess/trustProvider:TrustProvider';
