// @pulumi/aws@7.7.0/appconfig/deploymentStrategy.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeploymentStrategy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an AppConfig Deployment Strategy resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.appconfig.DeploymentStrategy("example", {
 *     name: "example-deployment-strategy-tf",
 *     description: "Example Deployment Strategy",
 *     deploymentDurationInMinutes: 3,
 *     finalBakeTimeInMinutes: 4,
 *     growthFactor: 10,
 *     growthType: "LINEAR",
 *     replicateTo: "NONE",
 *     tags: {
 *         Type: "AppConfig Deployment Strategy",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import AppConfig Deployment Strategies using their deployment strategy ID. For example:
 *
 * ```sh
 * $ pulumi import aws:appconfig/deploymentStrategy:DeploymentStrategy example 11xxxxx
 * ```
 */
class DeploymentStrategy extends pulumi.CustomResource {
    /**
     * Get an existing DeploymentStrategy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DeploymentStrategy(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DeploymentStrategy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeploymentStrategy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["deploymentDurationInMinutes"] = state?.deploymentDurationInMinutes;
            resourceInputs["description"] = state?.description;
            resourceInputs["finalBakeTimeInMinutes"] = state?.finalBakeTimeInMinutes;
            resourceInputs["growthFactor"] = state?.growthFactor;
            resourceInputs["growthType"] = state?.growthType;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["replicateTo"] = state?.replicateTo;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.deploymentDurationInMinutes === undefined && !opts.urn) {
                throw new Error("Missing required property 'deploymentDurationInMinutes'");
            }
            if (args?.growthFactor === undefined && !opts.urn) {
                throw new Error("Missing required property 'growthFactor'");
            }
            if (args?.replicateTo === undefined && !opts.urn) {
                throw new Error("Missing required property 'replicateTo'");
            }
            resourceInputs["deploymentDurationInMinutes"] = args?.deploymentDurationInMinutes;
            resourceInputs["description"] = args?.description;
            resourceInputs["finalBakeTimeInMinutes"] = args?.finalBakeTimeInMinutes;
            resourceInputs["growthFactor"] = args?.growthFactor;
            resourceInputs["growthType"] = args?.growthType;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["replicateTo"] = args?.replicateTo;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DeploymentStrategy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DeploymentStrategy = DeploymentStrategy;
/** @internal */
DeploymentStrategy.__pulumiType = 'aws:appconfig/deploymentStrategy:DeploymentStrategy';
