// @pulumi/aws@7.7.0/eks/identityProviderConfig.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityProviderConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an EKS Identity Provider Configuration.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.eks.IdentityProviderConfig("example", {
 *     clusterName: exampleAwsEksCluster.name,
 *     oidc: {
 *         clientId: "your client_id",
 *         identityProviderConfigName: "example",
 *         issuerUrl: "your issuer_url",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import EKS Identity Provider Configurations using the `cluster_name` and `identity_provider_config_name` separated by a colon (`:`). For example:
 *
 * ```sh
 * $ pulumi import aws:eks/identityProviderConfig:IdentityProviderConfig my_identity_provider_config my_cluster:my_identity_provider_config
 * ```
 */
class IdentityProviderConfig extends pulumi.CustomResource {
    /**
     * Get an existing IdentityProviderConfig resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new IdentityProviderConfig(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of IdentityProviderConfig.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === IdentityProviderConfig.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["clusterName"] = state?.clusterName;
            resourceInputs["oidc"] = state?.oidc;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.clusterName === undefined && !opts.urn) {
                throw new Error("Missing required property 'clusterName'");
            }
            if (args?.oidc === undefined && !opts.urn) {
                throw new Error("Missing required property 'oidc'");
            }
            resourceInputs["clusterName"] = args?.clusterName;
            resourceInputs["oidc"] = args?.oidc;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(IdentityProviderConfig.__pulumiType, name, resourceInputs, opts);
    }
}
exports.IdentityProviderConfig = IdentityProviderConfig;
/** @internal */
IdentityProviderConfig.__pulumiType = 'aws:eks/identityProviderConfig:IdentityProviderConfig';
