// @pulumi/aws@7.7.0/apigateway/resource.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Resource = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an API Gateway Resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const myDemoAPI = new aws.apigateway.RestApi("MyDemoAPI", {
 *     name: "MyDemoAPI",
 *     description: "This is my API for demonstration purposes",
 * });
 * const myDemoResource = new aws.apigateway.Resource("MyDemoResource", {
 *     restApi: myDemoAPI.id,
 *     parentId: myDemoAPI.rootResourceId,
 *     pathPart: "mydemoresource",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_api_gateway_resource` using `REST-API-ID/RESOURCE-ID`. For example:
 *
 * ```sh
 * $ pulumi import aws:apigateway/resource:Resource example 12345abcde/67890fghij
 * ```
 */
class Resource extends pulumi.CustomResource {
    /**
     * Get an existing Resource resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Resource(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Resource.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Resource.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["parentId"] = state?.parentId;
            resourceInputs["path"] = state?.path;
            resourceInputs["pathPart"] = state?.pathPart;
            resourceInputs["region"] = state?.region;
            resourceInputs["restApi"] = state?.restApi;
        }
        else {
            const args = argsOrState;
            if (args?.parentId === undefined && !opts.urn) {
                throw new Error("Missing required property 'parentId'");
            }
            if (args?.pathPart === undefined && !opts.urn) {
                throw new Error("Missing required property 'pathPart'");
            }
            if (args?.restApi === undefined && !opts.urn) {
                throw new Error("Missing required property 'restApi'");
            }
            resourceInputs["parentId"] = args?.parentId;
            resourceInputs["pathPart"] = args?.pathPart;
            resourceInputs["region"] = args?.region;
            resourceInputs["restApi"] = args?.restApi;
            resourceInputs["path"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Resource.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Resource = Resource;
/** @internal */
Resource.__pulumiType = 'aws:apigateway/resource:Resource';
