// @pulumi/aws@7.7.0/apigateway/response.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Response = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an API Gateway Gateway Response for a REST API Gateway.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const main = new aws.apigateway.RestApi("main", {name: "MyDemoAPI"});
 * const test = new aws.apigateway.Response("test", {
 *     restApiId: main.id,
 *     statusCode: "401",
 *     responseType: "UNAUTHORIZED",
 *     responseTemplates: {
 *         "application/json": "{\"message\":$context.error.messageString}",
 *     },
 *     responseParameters: {
 *         "gatewayresponse.header.Authorization": "'Basic'",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_api_gateway_gateway_response` using `REST-API-ID/RESPONSE-TYPE`. For example:
 *
 * ```sh
 * $ pulumi import aws:apigateway/response:Response example 12345abcde/UNAUTHORIZED
 * ```
 */
class Response extends pulumi.CustomResource {
    /**
     * Get an existing Response resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Response(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Response.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Response.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["region"] = state?.region;
            resourceInputs["responseParameters"] = state?.responseParameters;
            resourceInputs["responseTemplates"] = state?.responseTemplates;
            resourceInputs["responseType"] = state?.responseType;
            resourceInputs["restApiId"] = state?.restApiId;
            resourceInputs["statusCode"] = state?.statusCode;
        }
        else {
            const args = argsOrState;
            if (args?.responseType === undefined && !opts.urn) {
                throw new Error("Missing required property 'responseType'");
            }
            if (args?.restApiId === undefined && !opts.urn) {
                throw new Error("Missing required property 'restApiId'");
            }
            resourceInputs["region"] = args?.region;
            resourceInputs["responseParameters"] = args?.responseParameters;
            resourceInputs["responseTemplates"] = args?.responseTemplates;
            resourceInputs["responseType"] = args?.responseType;
            resourceInputs["restApiId"] = args?.restApiId;
            resourceInputs["statusCode"] = args?.statusCode;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Response.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Response = Response;
/** @internal */
Response.__pulumiType = 'aws:apigateway/response:Response';
