// @pulumi/aws@7.7.0/appintegrations/dataIntegration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataIntegration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Amazon AppIntegrations Data Integration resource.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.appintegrations.DataIntegration("example", {
 *     name: "example",
 *     description: "example",
 *     kmsKey: test.arn,
 *     sourceUri: "Salesforce://AppFlow/example",
 *     scheduleConfig: {
 *         firstExecutionFrom: "1439788442681",
 *         object: "Account",
 *         scheduleExpression: "rate(1 hour)",
 *     },
 *     tags: {
 *         Key1: "Value1",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Amazon AppIntegrations Data Integrations using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:appintegrations/dataIntegration:DataIntegration example 12345678-1234-1234-1234-123456789123
 * ```
 */
class DataIntegration extends pulumi.CustomResource {
    /**
     * Get an existing DataIntegration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DataIntegration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DataIntegration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DataIntegration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["description"] = state?.description;
            resourceInputs["kmsKey"] = state?.kmsKey;
            resourceInputs["name"] = state?.name;
            resourceInputs["region"] = state?.region;
            resourceInputs["scheduleConfig"] = state?.scheduleConfig;
            resourceInputs["sourceUri"] = state?.sourceUri;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.kmsKey === undefined && !opts.urn) {
                throw new Error("Missing required property 'kmsKey'");
            }
            if (args?.scheduleConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'scheduleConfig'");
            }
            if (args?.sourceUri === undefined && !opts.urn) {
                throw new Error("Missing required property 'sourceUri'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["kmsKey"] = args?.kmsKey;
            resourceInputs["name"] = args?.name;
            resourceInputs["region"] = args?.region;
            resourceInputs["scheduleConfig"] = args?.scheduleConfig;
            resourceInputs["sourceUri"] = args?.sourceUri;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DataIntegration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DataIntegration = DataIntegration;
/** @internal */
DataIntegration.__pulumiType = 'aws:appintegrations/dataIntegration:DataIntegration';
