// @pulumi/aws@7.7.0/servicecatalog/productPortfolioAssociation.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductPortfolioAssociation = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a Service Catalog Product Portfolio Association.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.servicecatalog.ProductPortfolioAssociation("example", {
 *     portfolioId: "port-68656c6c6f",
 *     productId: "prod-dnigbtea24ste",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_servicecatalog_product_portfolio_association` using the accept language, portfolio ID, and product ID. For example:
 *
 * ```sh
 * $ pulumi import aws:servicecatalog/productPortfolioAssociation:ProductPortfolioAssociation example en:port-68656c6c6f:prod-dnigbtea24ste
 * ```
 */
class ProductPortfolioAssociation extends pulumi.CustomResource {
    /**
     * Get an existing ProductPortfolioAssociation resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ProductPortfolioAssociation(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ProductPortfolioAssociation.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ProductPortfolioAssociation.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["acceptLanguage"] = state?.acceptLanguage;
            resourceInputs["portfolioId"] = state?.portfolioId;
            resourceInputs["productId"] = state?.productId;
            resourceInputs["region"] = state?.region;
            resourceInputs["sourcePortfolioId"] = state?.sourcePortfolioId;
        }
        else {
            const args = argsOrState;
            if (args?.portfolioId === undefined && !opts.urn) {
                throw new Error("Missing required property 'portfolioId'");
            }
            if (args?.productId === undefined && !opts.urn) {
                throw new Error("Missing required property 'productId'");
            }
            resourceInputs["acceptLanguage"] = args?.acceptLanguage;
            resourceInputs["portfolioId"] = args?.portfolioId;
            resourceInputs["productId"] = args?.productId;
            resourceInputs["region"] = args?.region;
            resourceInputs["sourcePortfolioId"] = args?.sourcePortfolioId;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ProductPortfolioAssociation.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ProductPortfolioAssociation = ProductPortfolioAssociation;
/** @internal */
ProductPortfolioAssociation.__pulumiType = 'aws:servicecatalog/productPortfolioAssociation:ProductPortfolioAssociation';
