// @pulumi/aws@7.7.0/s3/directoryBucket.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DirectoryBucket = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an Amazon S3 Express directory bucket resource.
 *
 * ## Example Usage
 *
 * ### Availability Zone
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.s3.DirectoryBucket("example", {
 *     bucket: "example--usw2-az1--x-s3",
 *     location: {
 *         name: "usw2-az1",
 *     },
 * });
 * ```
 *
 * ### Dedicated Local Zone
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const exampleLocalZone = new aws.s3.DirectoryBucket("example_local_zone", {
 *     bucket: "example--usw2-xxx-lz1--x-s3",
 *     location: {
 *         name: "usw2-xxx-lz1",
 *         type: "LocalZone",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import S3 bucket using `bucket`. For example:
 *
 * ```sh
 * $ pulumi import aws:s3/directoryBucket:DirectoryBucket example example--usw2-az1--x-s3
 * ```
 */
class DirectoryBucket extends pulumi.CustomResource {
    /**
     * Get an existing DirectoryBucket resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DirectoryBucket(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DirectoryBucket.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DirectoryBucket.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["bucket"] = state?.bucket;
            resourceInputs["dataRedundancy"] = state?.dataRedundancy;
            resourceInputs["forceDestroy"] = state?.forceDestroy;
            resourceInputs["location"] = state?.location;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            if (args?.bucket === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucket'");
            }
            resourceInputs["bucket"] = args?.bucket;
            resourceInputs["dataRedundancy"] = args?.dataRedundancy;
            resourceInputs["forceDestroy"] = args?.forceDestroy;
            resourceInputs["location"] = args?.location;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["type"] = args?.type;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DirectoryBucket.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DirectoryBucket = DirectoryBucket;
/** @internal */
DirectoryBucket.__pulumiType = 'aws:s3/directoryBucket:DirectoryBucket';
