// @pulumi/aws@7.7.0/s3/index.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCanonicalUserIdOutput = exports.getCanonicalUserId = exports.getBucketPolicyOutput = exports.getBucketPolicy = exports.getBucketObjectsOutput = exports.getBucketObjects = exports.getBucketObjectOutput = exports.getBucketObject = exports.getBucketOutput = exports.getBucket = exports.getAccountPublicAccessBlockOutput = exports.getAccountPublicAccessBlock = exports.getAccessPointOutput = exports.getAccessPoint = exports.DirectoryBucket = exports.BucketWebsiteConfigurationV2 = exports.BucketWebsiteConfiguration = exports.BucketVersioningV2 = exports.BucketVersioning = exports.BucketV2 = exports.BucketServerSideEncryptionConfigurationV2 = exports.BucketServerSideEncryptionConfiguration = exports.BucketRequestPaymentConfigurationV2 = exports.BucketRequestPaymentConfiguration = exports.BucketReplicationConfig = exports.BucketPublicAccessBlock = exports.BucketPolicy = exports.BucketOwnershipControls = exports.BucketObjectv2 = exports.BucketObjectLockConfigurationV2 = exports.BucketObjectLockConfiguration = exports.BucketObject = exports.BucketNotification = exports.BucketMetric = exports.BucketMetadataConfiguration = exports.BucketLoggingV2 = exports.BucketLogging = exports.BucketLifecycleConfigurationV2 = exports.BucketLifecycleConfiguration = exports.BucketIntelligentTieringConfiguration = exports.BucketCorsConfigurationV2 = exports.BucketCorsConfiguration = exports.BucketAclV2 = exports.BucketAcl = exports.BucketAccelerateConfigurationV2 = exports.BucketAccelerateConfiguration = exports.Bucket = exports.AnalyticsConfiguration = exports.AccountPublicAccessBlock = exports.AccessPoint = void 0;
exports.ObjectCopy = exports.Inventory = exports.getObjectsOutput = exports.getObjects = exports.getObjectOutput = exports.getObject = exports.getDirectoryBucketsOutput = exports.getDirectoryBuckets = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessPoint = null;
utilities.lazyLoad(exports, ["AccessPoint"], () => require("./accessPoint"));
exports.AccountPublicAccessBlock = null;
utilities.lazyLoad(exports, ["AccountPublicAccessBlock"], () => require("./accountPublicAccessBlock"));
exports.AnalyticsConfiguration = null;
utilities.lazyLoad(exports, ["AnalyticsConfiguration"], () => require("./analyticsConfiguration"));
exports.Bucket = null;
utilities.lazyLoad(exports, ["Bucket"], () => require("./bucket"));
exports.BucketAccelerateConfiguration = null;
utilities.lazyLoad(exports, ["BucketAccelerateConfiguration"], () => require("./bucketAccelerateConfiguration"));
exports.BucketAccelerateConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketAccelerateConfigurationV2"], () => require("./bucketAccelerateConfigurationV2"));
exports.BucketAcl = null;
utilities.lazyLoad(exports, ["BucketAcl"], () => require("./bucketAcl"));
exports.BucketAclV2 = null;
utilities.lazyLoad(exports, ["BucketAclV2"], () => require("./bucketAclV2"));
exports.BucketCorsConfiguration = null;
utilities.lazyLoad(exports, ["BucketCorsConfiguration"], () => require("./bucketCorsConfiguration"));
exports.BucketCorsConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketCorsConfigurationV2"], () => require("./bucketCorsConfigurationV2"));
exports.BucketIntelligentTieringConfiguration = null;
utilities.lazyLoad(exports, ["BucketIntelligentTieringConfiguration"], () => require("./bucketIntelligentTieringConfiguration"));
exports.BucketLifecycleConfiguration = null;
utilities.lazyLoad(exports, ["BucketLifecycleConfiguration"], () => require("./bucketLifecycleConfiguration"));
exports.BucketLifecycleConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketLifecycleConfigurationV2"], () => require("./bucketLifecycleConfigurationV2"));
exports.BucketLogging = null;
utilities.lazyLoad(exports, ["BucketLogging"], () => require("./bucketLogging"));
exports.BucketLoggingV2 = null;
utilities.lazyLoad(exports, ["BucketLoggingV2"], () => require("./bucketLoggingV2"));
exports.BucketMetadataConfiguration = null;
utilities.lazyLoad(exports, ["BucketMetadataConfiguration"], () => require("./bucketMetadataConfiguration"));
exports.BucketMetric = null;
utilities.lazyLoad(exports, ["BucketMetric"], () => require("./bucketMetric"));
exports.BucketNotification = null;
utilities.lazyLoad(exports, ["BucketNotification"], () => require("./bucketNotification"));
exports.BucketObject = null;
utilities.lazyLoad(exports, ["BucketObject"], () => require("./bucketObject"));
exports.BucketObjectLockConfiguration = null;
utilities.lazyLoad(exports, ["BucketObjectLockConfiguration"], () => require("./bucketObjectLockConfiguration"));
exports.BucketObjectLockConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketObjectLockConfigurationV2"], () => require("./bucketObjectLockConfigurationV2"));
exports.BucketObjectv2 = null;
utilities.lazyLoad(exports, ["BucketObjectv2"], () => require("./bucketObjectv2"));
exports.BucketOwnershipControls = null;
utilities.lazyLoad(exports, ["BucketOwnershipControls"], () => require("./bucketOwnershipControls"));
exports.BucketPolicy = null;
utilities.lazyLoad(exports, ["BucketPolicy"], () => require("./bucketPolicy"));
exports.BucketPublicAccessBlock = null;
utilities.lazyLoad(exports, ["BucketPublicAccessBlock"], () => require("./bucketPublicAccessBlock"));
exports.BucketReplicationConfig = null;
utilities.lazyLoad(exports, ["BucketReplicationConfig"], () => require("./bucketReplicationConfig"));
exports.BucketRequestPaymentConfiguration = null;
utilities.lazyLoad(exports, ["BucketRequestPaymentConfiguration"], () => require("./bucketRequestPaymentConfiguration"));
exports.BucketRequestPaymentConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketRequestPaymentConfigurationV2"], () => require("./bucketRequestPaymentConfigurationV2"));
exports.BucketServerSideEncryptionConfiguration = null;
utilities.lazyLoad(exports, ["BucketServerSideEncryptionConfiguration"], () => require("./bucketServerSideEncryptionConfiguration"));
exports.BucketServerSideEncryptionConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketServerSideEncryptionConfigurationV2"], () => require("./bucketServerSideEncryptionConfigurationV2"));
exports.BucketV2 = null;
utilities.lazyLoad(exports, ["BucketV2"], () => require("./bucketV2"));
exports.BucketVersioning = null;
utilities.lazyLoad(exports, ["BucketVersioning"], () => require("./bucketVersioning"));
exports.BucketVersioningV2 = null;
utilities.lazyLoad(exports, ["BucketVersioningV2"], () => require("./bucketVersioningV2"));
exports.BucketWebsiteConfiguration = null;
utilities.lazyLoad(exports, ["BucketWebsiteConfiguration"], () => require("./bucketWebsiteConfiguration"));
exports.BucketWebsiteConfigurationV2 = null;
utilities.lazyLoad(exports, ["BucketWebsiteConfigurationV2"], () => require("./bucketWebsiteConfigurationV2"));
exports.DirectoryBucket = null;
utilities.lazyLoad(exports, ["DirectoryBucket"], () => require("./directoryBucket"));
exports.getAccessPoint = null;
exports.getAccessPointOutput = null;
utilities.lazyLoad(exports, ["getAccessPoint", "getAccessPointOutput"], () => require("./getAccessPoint"));
exports.getAccountPublicAccessBlock = null;
exports.getAccountPublicAccessBlockOutput = null;
utilities.lazyLoad(exports, ["getAccountPublicAccessBlock", "getAccountPublicAccessBlockOutput"], () => require("./getAccountPublicAccessBlock"));
exports.getBucket = null;
exports.getBucketOutput = null;
utilities.lazyLoad(exports, ["getBucket", "getBucketOutput"], () => require("./getBucket"));
exports.getBucketObject = null;
exports.getBucketObjectOutput = null;
utilities.lazyLoad(exports, ["getBucketObject", "getBucketObjectOutput"], () => require("./getBucketObject"));
exports.getBucketObjects = null;
exports.getBucketObjectsOutput = null;
utilities.lazyLoad(exports, ["getBucketObjects", "getBucketObjectsOutput"], () => require("./getBucketObjects"));
exports.getBucketPolicy = null;
exports.getBucketPolicyOutput = null;
utilities.lazyLoad(exports, ["getBucketPolicy", "getBucketPolicyOutput"], () => require("./getBucketPolicy"));
exports.getCanonicalUserId = null;
exports.getCanonicalUserIdOutput = null;
utilities.lazyLoad(exports, ["getCanonicalUserId", "getCanonicalUserIdOutput"], () => require("./getCanonicalUserId"));
exports.getDirectoryBuckets = null;
exports.getDirectoryBucketsOutput = null;
utilities.lazyLoad(exports, ["getDirectoryBuckets", "getDirectoryBucketsOutput"], () => require("./getDirectoryBuckets"));
exports.getObject = null;
exports.getObjectOutput = null;
utilities.lazyLoad(exports, ["getObject", "getObjectOutput"], () => require("./getObject"));
exports.getObjects = null;
exports.getObjectsOutput = null;
utilities.lazyLoad(exports, ["getObjects", "getObjectsOutput"], () => require("./getObjects"));
exports.Inventory = null;
utilities.lazyLoad(exports, ["Inventory"], () => require("./inventory"));
exports.ObjectCopy = null;
utilities.lazyLoad(exports, ["ObjectCopy"], () => require("./objectCopy"));
__exportStar(require("./routingRules"), exports);
__exportStar(require("./s3Mixins"), exports);
// Export enums:
__exportStar(require("../types/enums/s3"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:s3/accessPoint:AccessPoint":
                return new exports.AccessPoint(name, undefined, { urn });
            case "aws:s3/accountPublicAccessBlock:AccountPublicAccessBlock":
                return new exports.AccountPublicAccessBlock(name, undefined, { urn });
            case "aws:s3/analyticsConfiguration:AnalyticsConfiguration":
                return new exports.AnalyticsConfiguration(name, undefined, { urn });
            case "aws:s3/bucket:Bucket":
                return new exports.Bucket(name, undefined, { urn });
            case "aws:s3/bucketAccelerateConfiguration:BucketAccelerateConfiguration":
                return new exports.BucketAccelerateConfiguration(name, undefined, { urn });
            case "aws:s3/bucketAccelerateConfigurationV2:BucketAccelerateConfigurationV2":
                return new exports.BucketAccelerateConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketAcl:BucketAcl":
                return new exports.BucketAcl(name, undefined, { urn });
            case "aws:s3/bucketAclV2:BucketAclV2":
                return new exports.BucketAclV2(name, undefined, { urn });
            case "aws:s3/bucketCorsConfiguration:BucketCorsConfiguration":
                return new exports.BucketCorsConfiguration(name, undefined, { urn });
            case "aws:s3/bucketCorsConfigurationV2:BucketCorsConfigurationV2":
                return new exports.BucketCorsConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketIntelligentTieringConfiguration:BucketIntelligentTieringConfiguration":
                return new exports.BucketIntelligentTieringConfiguration(name, undefined, { urn });
            case "aws:s3/bucketLifecycleConfiguration:BucketLifecycleConfiguration":
                return new exports.BucketLifecycleConfiguration(name, undefined, { urn });
            case "aws:s3/bucketLifecycleConfigurationV2:BucketLifecycleConfigurationV2":
                return new exports.BucketLifecycleConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketLogging:BucketLogging":
                return new exports.BucketLogging(name, undefined, { urn });
            case "aws:s3/bucketLoggingV2:BucketLoggingV2":
                return new exports.BucketLoggingV2(name, undefined, { urn });
            case "aws:s3/bucketMetadataConfiguration:BucketMetadataConfiguration":
                return new exports.BucketMetadataConfiguration(name, undefined, { urn });
            case "aws:s3/bucketMetric:BucketMetric":
                return new exports.BucketMetric(name, undefined, { urn });
            case "aws:s3/bucketNotification:BucketNotification":
                return new exports.BucketNotification(name, undefined, { urn });
            case "aws:s3/bucketObject:BucketObject":
                return new exports.BucketObject(name, undefined, { urn });
            case "aws:s3/bucketObjectLockConfiguration:BucketObjectLockConfiguration":
                return new exports.BucketObjectLockConfiguration(name, undefined, { urn });
            case "aws:s3/bucketObjectLockConfigurationV2:BucketObjectLockConfigurationV2":
                return new exports.BucketObjectLockConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketObjectv2:BucketObjectv2":
                return new exports.BucketObjectv2(name, undefined, { urn });
            case "aws:s3/bucketOwnershipControls:BucketOwnershipControls":
                return new exports.BucketOwnershipControls(name, undefined, { urn });
            case "aws:s3/bucketPolicy:BucketPolicy":
                return new exports.BucketPolicy(name, undefined, { urn });
            case "aws:s3/bucketPublicAccessBlock:BucketPublicAccessBlock":
                return new exports.BucketPublicAccessBlock(name, undefined, { urn });
            case "aws:s3/bucketReplicationConfig:BucketReplicationConfig":
                return new exports.BucketReplicationConfig(name, undefined, { urn });
            case "aws:s3/bucketRequestPaymentConfiguration:BucketRequestPaymentConfiguration":
                return new exports.BucketRequestPaymentConfiguration(name, undefined, { urn });
            case "aws:s3/bucketRequestPaymentConfigurationV2:BucketRequestPaymentConfigurationV2":
                return new exports.BucketRequestPaymentConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketServerSideEncryptionConfiguration:BucketServerSideEncryptionConfiguration":
                return new exports.BucketServerSideEncryptionConfiguration(name, undefined, { urn });
            case "aws:s3/bucketServerSideEncryptionConfigurationV2:BucketServerSideEncryptionConfigurationV2":
                return new exports.BucketServerSideEncryptionConfigurationV2(name, undefined, { urn });
            case "aws:s3/bucketV2:BucketV2":
                return new exports.BucketV2(name, undefined, { urn });
            case "aws:s3/bucketVersioning:BucketVersioning":
                return new exports.BucketVersioning(name, undefined, { urn });
            case "aws:s3/bucketVersioningV2:BucketVersioningV2":
                return new exports.BucketVersioningV2(name, undefined, { urn });
            case "aws:s3/bucketWebsiteConfiguration:BucketWebsiteConfiguration":
                return new exports.BucketWebsiteConfiguration(name, undefined, { urn });
            case "aws:s3/bucketWebsiteConfigurationV2:BucketWebsiteConfigurationV2":
                return new exports.BucketWebsiteConfigurationV2(name, undefined, { urn });
            case "aws:s3/directoryBucket:DirectoryBucket":
                return new exports.DirectoryBucket(name, undefined, { urn });
            case "aws:s3/inventory:Inventory":
                return new exports.Inventory(name, undefined, { urn });
            case "aws:s3/objectCopy:ObjectCopy":
                return new exports.ObjectCopy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "s3/accessPoint", _module);
pulumi.runtime.registerResourceModule("aws", "s3/accountPublicAccessBlock", _module);
pulumi.runtime.registerResourceModule("aws", "s3/analyticsConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucket", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketAccelerateConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketAccelerateConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketAcl", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketAclV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketCorsConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketCorsConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketIntelligentTieringConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketLifecycleConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketLifecycleConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketLogging", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketLoggingV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketMetadataConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketMetric", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketNotification", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObject", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObjectLockConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObjectLockConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketObjectv2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketOwnershipControls", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketPolicy", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketPublicAccessBlock", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketReplicationConfig", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketRequestPaymentConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketRequestPaymentConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketServerSideEncryptionConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketServerSideEncryptionConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketVersioning", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketVersioningV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketWebsiteConfiguration", _module);
pulumi.runtime.registerResourceModule("aws", "s3/bucketWebsiteConfigurationV2", _module);
pulumi.runtime.registerResourceModule("aws", "s3/directoryBucket", _module);
pulumi.runtime.registerResourceModule("aws", "s3/inventory", _module);
pulumi.runtime.registerResourceModule("aws", "s3/objectCopy", _module);
