// @pulumi/aws@7.7.0/qldb/ledger.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Ledger = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides an AWS Quantum Ledger Database (QLDB) resource
 *
 * > **NOTE:** Deletion protection is enabled by default. To successfully delete this resource via this provider, `deletionProtection = false` must be applied before attempting deletion.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const sample_ledger = new aws.qldb.Ledger("sample-ledger", {
 *     name: "sample-ledger",
 *     permissionsMode: "STANDARD",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import QLDB Ledgers using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:qldb/ledger:Ledger sample-ledger sample-ledger
 * ```
 */
class Ledger extends pulumi.CustomResource {
    /**
     * Get an existing Ledger resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Ledger(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Ledger.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Ledger.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["deletionProtection"] = state?.deletionProtection;
            resourceInputs["kmsKey"] = state?.kmsKey;
            resourceInputs["name"] = state?.name;
            resourceInputs["permissionsMode"] = state?.permissionsMode;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.permissionsMode === undefined && !opts.urn) {
                throw new Error("Missing required property 'permissionsMode'");
            }
            resourceInputs["deletionProtection"] = args?.deletionProtection;
            resourceInputs["kmsKey"] = args?.kmsKey;
            resourceInputs["name"] = args?.name;
            resourceInputs["permissionsMode"] = args?.permissionsMode;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Ledger.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Ledger = Ledger;
/** @internal */
Ledger.__pulumiType = 'aws:qldb/ledger:Ledger';
