// @pulumi/aws@7.7.0/apprunner/autoScalingConfigurationVersion.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutoScalingConfigurationVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an App Runner AutoScaling Configuration Version.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.apprunner.AutoScalingConfigurationVersion("example", {
 *     autoScalingConfigurationName: "example",
 *     maxConcurrency: 50,
 *     maxSize: 10,
 *     minSize: 2,
 *     tags: {
 *         Name: "example-apprunner-autoscaling",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import App Runner AutoScaling Configuration Versions using the `arn`. For example:
 *
 * ```sh
 * $ pulumi import aws:apprunner/autoScalingConfigurationVersion:AutoScalingConfigurationVersion example "arn:aws:apprunner:us-east-1:1234567890:autoscalingconfiguration/example/1/69bdfe0115224b0db49398b7beb68e0f
 * ```
 */
class AutoScalingConfigurationVersion extends pulumi.CustomResource {
    /**
     * Get an existing AutoScalingConfigurationVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AutoScalingConfigurationVersion(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of AutoScalingConfigurationVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AutoScalingConfigurationVersion.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["autoScalingConfigurationName"] = state?.autoScalingConfigurationName;
            resourceInputs["autoScalingConfigurationRevision"] = state?.autoScalingConfigurationRevision;
            resourceInputs["hasAssociatedService"] = state?.hasAssociatedService;
            resourceInputs["isDefault"] = state?.isDefault;
            resourceInputs["latest"] = state?.latest;
            resourceInputs["maxConcurrency"] = state?.maxConcurrency;
            resourceInputs["maxSize"] = state?.maxSize;
            resourceInputs["minSize"] = state?.minSize;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.autoScalingConfigurationName === undefined && !opts.urn) {
                throw new Error("Missing required property 'autoScalingConfigurationName'");
            }
            resourceInputs["autoScalingConfigurationName"] = args?.autoScalingConfigurationName;
            resourceInputs["maxConcurrency"] = args?.maxConcurrency;
            resourceInputs["maxSize"] = args?.maxSize;
            resourceInputs["minSize"] = args?.minSize;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["autoScalingConfigurationRevision"] = undefined /*out*/;
            resourceInputs["hasAssociatedService"] = undefined /*out*/;
            resourceInputs["isDefault"] = undefined /*out*/;
            resourceInputs["latest"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AutoScalingConfigurationVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AutoScalingConfigurationVersion = AutoScalingConfigurationVersion;
/** @internal */
AutoScalingConfigurationVersion.__pulumiType = 'aws:apprunner/autoScalingConfigurationVersion:AutoScalingConfigurationVersion';
