// @pulumi/aws@7.7.0/quicksight/analysis.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Analysis = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing a QuickSight Analysis.
 *
 * ## Example Usage
 *
 * ### From Source Template
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.quicksight.Analysis("example", {
 *     analysisId: "example-id",
 *     name: "example-name",
 *     sourceEntity: {
 *         sourceTemplate: {
 *             arn: source.arn,
 *             dataSetReferences: [{
 *                 dataSetArn: dataset.arn,
 *                 dataSetPlaceholder: "1",
 *             }],
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import a QuickSight Analysis using the AWS account ID and analysis ID separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:quicksight/analysis:Analysis example 123456789012,example-id
 * ```
 */
class Analysis extends pulumi.CustomResource {
    /**
     * Get an existing Analysis resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Analysis(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Analysis.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Analysis.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["analysisId"] = state?.analysisId;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["awsAccountId"] = state?.awsAccountId;
            resourceInputs["createdTime"] = state?.createdTime;
            resourceInputs["lastPublishedTime"] = state?.lastPublishedTime;
            resourceInputs["lastUpdatedTime"] = state?.lastUpdatedTime;
            resourceInputs["name"] = state?.name;
            resourceInputs["parameters"] = state?.parameters;
            resourceInputs["permissions"] = state?.permissions;
            resourceInputs["recoveryWindowInDays"] = state?.recoveryWindowInDays;
            resourceInputs["region"] = state?.region;
            resourceInputs["sourceEntity"] = state?.sourceEntity;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["themeArn"] = state?.themeArn;
        }
        else {
            const args = argsOrState;
            if (args?.analysisId === undefined && !opts.urn) {
                throw new Error("Missing required property 'analysisId'");
            }
            resourceInputs["analysisId"] = args?.analysisId;
            resourceInputs["awsAccountId"] = args?.awsAccountId;
            resourceInputs["name"] = args?.name;
            resourceInputs["parameters"] = args?.parameters;
            resourceInputs["permissions"] = args?.permissions;
            resourceInputs["recoveryWindowInDays"] = args?.recoveryWindowInDays;
            resourceInputs["region"] = args?.region;
            resourceInputs["sourceEntity"] = args?.sourceEntity;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["themeArn"] = args?.themeArn;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["lastPublishedTime"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Analysis.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Analysis = Analysis;
/** @internal */
Analysis.__pulumiType = 'aws:quicksight/analysis:Analysis';
