// @pulumi/aws@7.7.0/quicksight/theme.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Theme = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Resource for managing a QuickSight Theme.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.quicksight.Theme("example", {
 *     themeId: "example",
 *     name: "example",
 *     baseThemeId: "MIDNIGHT",
 *     configuration: {
 *         dataColorPalette: {
 *             colors: [
 *                 "#FFFFFF",
 *                 "#111111",
 *                 "#222222",
 *                 "#333333",
 *                 "#444444",
 *                 "#555555",
 *                 "#666666",
 *                 "#777777",
 *                 "#888888",
 *                 "#999999",
 *             ],
 *             emptyFillColor: "#FFFFFF",
 *             minMaxGradients: [
 *                 "#FFFFFF",
 *                 "#111111",
 *             ],
 *         },
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import a QuickSight Theme using the AWS account ID and theme ID separated by a comma (`,`). For example:
 *
 * ```sh
 * $ pulumi import aws:quicksight/theme:Theme example 123456789012,example-id
 * ```
 */
class Theme extends pulumi.CustomResource {
    /**
     * Get an existing Theme resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Theme(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Theme.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Theme.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["awsAccountId"] = state?.awsAccountId;
            resourceInputs["baseThemeId"] = state?.baseThemeId;
            resourceInputs["configuration"] = state?.configuration;
            resourceInputs["createdTime"] = state?.createdTime;
            resourceInputs["lastUpdatedTime"] = state?.lastUpdatedTime;
            resourceInputs["name"] = state?.name;
            resourceInputs["permissions"] = state?.permissions;
            resourceInputs["region"] = state?.region;
            resourceInputs["status"] = state?.status;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["themeId"] = state?.themeId;
            resourceInputs["versionDescription"] = state?.versionDescription;
            resourceInputs["versionNumber"] = state?.versionNumber;
        }
        else {
            const args = argsOrState;
            if (args?.baseThemeId === undefined && !opts.urn) {
                throw new Error("Missing required property 'baseThemeId'");
            }
            if (args?.themeId === undefined && !opts.urn) {
                throw new Error("Missing required property 'themeId'");
            }
            resourceInputs["awsAccountId"] = args?.awsAccountId;
            resourceInputs["baseThemeId"] = args?.baseThemeId;
            resourceInputs["configuration"] = args?.configuration;
            resourceInputs["name"] = args?.name;
            resourceInputs["permissions"] = args?.permissions;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["themeId"] = args?.themeId;
            resourceInputs["versionDescription"] = args?.versionDescription;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["createdTime"] = undefined /*out*/;
            resourceInputs["lastUpdatedTime"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["versionNumber"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Theme.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Theme = Theme;
/** @internal */
Theme.__pulumiType = 'aws:quicksight/theme:Theme';
