// @pulumi/aws@7.7.0/quicksight/userCustomPermission.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserCustomPermission = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages the custom permissions profile for a user.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.quicksight.UserCustomPermission("example", {
 *     userName: exampleAwsQuicksightUser.userName,
 *     customPermissionsName: exampleAwsQuicksightCustomPermissions.customPermissionsName,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import QuickSight user custom permissions using a comma-delimited string combining the `aws_account_id`, `namespace`, and `user_name`. For example:
 *
 * ```sh
 * $ pulumi import aws:quicksight/userCustomPermission:UserCustomPermission example 012345678901,default,user1
 * ```
 */
class UserCustomPermission extends pulumi.CustomResource {
    /**
     * Get an existing UserCustomPermission resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new UserCustomPermission(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of UserCustomPermission.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === UserCustomPermission.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["awsAccountId"] = state?.awsAccountId;
            resourceInputs["customPermissionsName"] = state?.customPermissionsName;
            resourceInputs["namespace"] = state?.namespace;
            resourceInputs["region"] = state?.region;
            resourceInputs["userName"] = state?.userName;
        }
        else {
            const args = argsOrState;
            if (args?.customPermissionsName === undefined && !opts.urn) {
                throw new Error("Missing required property 'customPermissionsName'");
            }
            if (args?.userName === undefined && !opts.urn) {
                throw new Error("Missing required property 'userName'");
            }
            resourceInputs["awsAccountId"] = args?.awsAccountId;
            resourceInputs["customPermissionsName"] = args?.customPermissionsName;
            resourceInputs["namespace"] = args?.namespace;
            resourceInputs["region"] = args?.region;
            resourceInputs["userName"] = args?.userName;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(UserCustomPermission.__pulumiType, name, resourceInputs, opts);
    }
}
exports.UserCustomPermission = UserCustomPermission;
/** @internal */
UserCustomPermission.__pulumiType = 'aws:quicksight/userCustomPermission:UserCustomPermission';
