// @pulumi/aws@7.7.0/sagemaker/deviceFleet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceFleet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a SageMaker AI Device Fleet resource.
 *
 * ## Example Usage
 *
 * ### Basic usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.DeviceFleet("example", {
 *     deviceFleetName: "example",
 *     roleArn: test.arn,
 *     outputConfig: {
 *         s3OutputLocation: `s3://${exampleAwsS3Bucket.bucket}/prefix/`,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import SageMaker AI Device Fleets using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/deviceFleet:DeviceFleet example my-fleet
 * ```
 */
class DeviceFleet extends pulumi.CustomResource {
    /**
     * Get an existing DeviceFleet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DeviceFleet(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DeviceFleet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DeviceFleet.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["description"] = state?.description;
            resourceInputs["deviceFleetName"] = state?.deviceFleetName;
            resourceInputs["enableIotRoleAlias"] = state?.enableIotRoleAlias;
            resourceInputs["iotRoleAlias"] = state?.iotRoleAlias;
            resourceInputs["outputConfig"] = state?.outputConfig;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.deviceFleetName === undefined && !opts.urn) {
                throw new Error("Missing required property 'deviceFleetName'");
            }
            if (args?.outputConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'outputConfig'");
            }
            if (args?.roleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["description"] = args?.description;
            resourceInputs["deviceFleetName"] = args?.deviceFleetName;
            resourceInputs["enableIotRoleAlias"] = args?.enableIotRoleAlias;
            resourceInputs["outputConfig"] = args?.outputConfig;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["iotRoleAlias"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DeviceFleet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DeviceFleet = DeviceFleet;
/** @internal */
DeviceFleet.__pulumiType = 'aws:sagemaker/deviceFleet:DeviceFleet';
