// @pulumi/aws@7.7.0/sagemaker/endpointConfiguration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointConfiguration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a SageMaker AI endpoint configuration resource.
 *
 * ## Example Usage
 *
 * Basic usage:
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const ec = new aws.sagemaker.EndpointConfiguration("ec", {
 *     name: "my-endpoint-config",
 *     productionVariants: [{
 *         variantName: "variant-1",
 *         modelName: m.name,
 *         initialInstanceCount: 1,
 *         instanceType: "ml.t2.medium",
 *     }],
 *     tags: {
 *         Name: "foo",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import endpoint configurations using the `name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/endpointConfiguration:EndpointConfiguration test_endpoint_config endpoint-config-foo
 * ```
 */
class EndpointConfiguration extends pulumi.CustomResource {
    /**
     * Get an existing EndpointConfiguration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EndpointConfiguration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EndpointConfiguration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointConfiguration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["asyncInferenceConfig"] = state?.asyncInferenceConfig;
            resourceInputs["dataCaptureConfig"] = state?.dataCaptureConfig;
            resourceInputs["kmsKeyArn"] = state?.kmsKeyArn;
            resourceInputs["name"] = state?.name;
            resourceInputs["namePrefix"] = state?.namePrefix;
            resourceInputs["productionVariants"] = state?.productionVariants;
            resourceInputs["region"] = state?.region;
            resourceInputs["shadowProductionVariants"] = state?.shadowProductionVariants;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.productionVariants === undefined && !opts.urn) {
                throw new Error("Missing required property 'productionVariants'");
            }
            resourceInputs["asyncInferenceConfig"] = args?.asyncInferenceConfig;
            resourceInputs["dataCaptureConfig"] = args?.dataCaptureConfig;
            resourceInputs["kmsKeyArn"] = args?.kmsKeyArn;
            resourceInputs["name"] = args?.name;
            resourceInputs["namePrefix"] = args?.namePrefix;
            resourceInputs["productionVariants"] = args?.productionVariants;
            resourceInputs["region"] = args?.region;
            resourceInputs["shadowProductionVariants"] = args?.shadowProductionVariants;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EndpointConfiguration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EndpointConfiguration = EndpointConfiguration;
/** @internal */
EndpointConfiguration.__pulumiType = 'aws:sagemaker/endpointConfiguration:EndpointConfiguration';
