// @pulumi/aws@7.7.0/sagemaker/flowDefinition.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowDefinition = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a SageMaker AI Flow Definition resource.
 *
 * ## Example Usage
 *
 * ### Basic Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.FlowDefinition("example", {
 *     flowDefinitionName: "example",
 *     roleArn: exampleAwsIamRole.arn,
 *     humanLoopConfig: {
 *         humanTaskUiArn: exampleAwsSagemakerHumanTaskUi.arn,
 *         taskAvailabilityLifetimeInSeconds: 1,
 *         taskCount: 1,
 *         taskDescription: "example",
 *         taskTitle: "example",
 *         workteamArn: exampleAwsSagemakerWorkteam.arn,
 *     },
 *     outputConfig: {
 *         s3OutputPath: `s3://${exampleAwsS3Bucket.bucket}/`,
 *     },
 * });
 * ```
 *
 * ### Public Workteam Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.FlowDefinition("example", {
 *     flowDefinitionName: "example",
 *     roleArn: exampleAwsIamRole.arn,
 *     humanLoopConfig: {
 *         humanTaskUiArn: exampleAwsSagemakerHumanTaskUi.arn,
 *         taskAvailabilityLifetimeInSeconds: 1,
 *         taskCount: 1,
 *         taskDescription: "example",
 *         taskTitle: "example",
 *         workteamArn: `arn:aws:sagemaker:${current.region}:394669845002:workteam/public-crowd/default`,
 *         publicWorkforceTaskPrice: {
 *             amountInUsd: {
 *                 cents: 1,
 *                 tenthFractionsOfACent: 2,
 *             },
 *         },
 *     },
 *     outputConfig: {
 *         s3OutputPath: `s3://${exampleAwsS3Bucket.bucket}/`,
 *     },
 * });
 * ```
 *
 * ### Human Loop Activation Config Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.FlowDefinition("example", {
 *     flowDefinitionName: "example",
 *     roleArn: exampleAwsIamRole.arn,
 *     humanLoopConfig: {
 *         humanTaskUiArn: exampleAwsSagemakerHumanTaskUi.arn,
 *         taskAvailabilityLifetimeInSeconds: 1,
 *         taskCount: 1,
 *         taskDescription: "example",
 *         taskTitle: "example",
 *         workteamArn: exampleAwsSagemakerWorkteam.arn,
 *     },
 *     humanLoopRequestSource: {
 *         awsManagedHumanLoopRequestSource: "AWS/Textract/AnalyzeDocument/Forms/V1",
 *     },
 *     humanLoopActivationConfig: {
 *         humanLoopActivationConditionsConfig: {
 *             humanLoopActivationConditions: `        {
 * \x09\x09\x09"Conditions": [
 * \x09\x09\x09  {
 * \x09\x09\x09\x09"ConditionType": "Sampling",
 * \x09\x09\x09\x09"ConditionParameters": {
 * \x09\x09\x09\x09  "RandomSamplingPercentage": 5
 * \x09\x09\x09\x09}
 * \x09\x09\x09  }
 * \x09\x09\x09]
 * \x09\x09}
 * `,
 *         },
 *     },
 *     outputConfig: {
 *         s3OutputPath: `s3://${exampleAwsS3Bucket.bucket}/`,
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import SageMaker AI Flow Definitions using the `flow_definition_name`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/flowDefinition:FlowDefinition example example
 * ```
 */
class FlowDefinition extends pulumi.CustomResource {
    /**
     * Get an existing FlowDefinition resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new FlowDefinition(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of FlowDefinition.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === FlowDefinition.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["flowDefinitionName"] = state?.flowDefinitionName;
            resourceInputs["humanLoopActivationConfig"] = state?.humanLoopActivationConfig;
            resourceInputs["humanLoopConfig"] = state?.humanLoopConfig;
            resourceInputs["humanLoopRequestSource"] = state?.humanLoopRequestSource;
            resourceInputs["outputConfig"] = state?.outputConfig;
            resourceInputs["region"] = state?.region;
            resourceInputs["roleArn"] = state?.roleArn;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.flowDefinitionName === undefined && !opts.urn) {
                throw new Error("Missing required property 'flowDefinitionName'");
            }
            if (args?.humanLoopConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'humanLoopConfig'");
            }
            if (args?.outputConfig === undefined && !opts.urn) {
                throw new Error("Missing required property 'outputConfig'");
            }
            if (args?.roleArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'roleArn'");
            }
            resourceInputs["flowDefinitionName"] = args?.flowDefinitionName;
            resourceInputs["humanLoopActivationConfig"] = args?.humanLoopActivationConfig;
            resourceInputs["humanLoopConfig"] = args?.humanLoopConfig;
            resourceInputs["humanLoopRequestSource"] = args?.humanLoopRequestSource;
            resourceInputs["outputConfig"] = args?.outputConfig;
            resourceInputs["region"] = args?.region;
            resourceInputs["roleArn"] = args?.roleArn;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(FlowDefinition.__pulumiType, name, resourceInputs, opts);
    }
}
exports.FlowDefinition = FlowDefinition;
/** @internal */
FlowDefinition.__pulumiType = 'aws:sagemaker/flowDefinition:FlowDefinition';
