// @pulumi/aws@7.7.0/sagemaker/space.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Space = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Provides a SageMaker AI Space resource.
 *
 * ## Example Usage
 *
 * ### Basic usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.sagemaker.Space("example", {
 *     domainId: test.id,
 *     spaceName: "example",
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import SageMaker AI Spaces using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:sagemaker/space:Space test_space arn:aws:sagemaker:us-west-2:123456789012:space/domain-id/space-name
 * ```
 */
class Space extends pulumi.CustomResource {
    /**
     * Get an existing Space resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Space(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Space.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Space.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["domainId"] = state?.domainId;
            resourceInputs["homeEfsFileSystemUid"] = state?.homeEfsFileSystemUid;
            resourceInputs["ownershipSettings"] = state?.ownershipSettings;
            resourceInputs["region"] = state?.region;
            resourceInputs["spaceDisplayName"] = state?.spaceDisplayName;
            resourceInputs["spaceName"] = state?.spaceName;
            resourceInputs["spaceSettings"] = state?.spaceSettings;
            resourceInputs["spaceSharingSettings"] = state?.spaceSharingSettings;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["url"] = state?.url;
        }
        else {
            const args = argsOrState;
            if (args?.domainId === undefined && !opts.urn) {
                throw new Error("Missing required property 'domainId'");
            }
            if (args?.spaceName === undefined && !opts.urn) {
                throw new Error("Missing required property 'spaceName'");
            }
            resourceInputs["domainId"] = args?.domainId;
            resourceInputs["ownershipSettings"] = args?.ownershipSettings;
            resourceInputs["region"] = args?.region;
            resourceInputs["spaceDisplayName"] = args?.spaceDisplayName;
            resourceInputs["spaceName"] = args?.spaceName;
            resourceInputs["spaceSettings"] = args?.spaceSettings;
            resourceInputs["spaceSharingSettings"] = args?.spaceSharingSettings;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["homeEfsFileSystemUid"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["url"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Space.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Space = Space;
/** @internal */
Space.__pulumiType = 'aws:sagemaker/space:Space';
