// @pulumi/aws@7.7.0/redshift/snapshotSchedule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SnapshotSchedule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const _default = new aws.redshift.SnapshotSchedule("default", {
 *     identifier: "tf-redshift-snapshot-schedule",
 *     definitions: ["rate(12 hours)"],
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import Redshift Snapshot Schedule using the `identifier`. For example:
 *
 * ```sh
 * $ pulumi import aws:redshift/snapshotSchedule:SnapshotSchedule default tf-redshift-snapshot-schedule
 * ```
 */
class SnapshotSchedule extends pulumi.CustomResource {
    /**
     * Get an existing SnapshotSchedule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SnapshotSchedule(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of SnapshotSchedule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SnapshotSchedule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["definitions"] = state?.definitions;
            resourceInputs["description"] = state?.description;
            resourceInputs["forceDestroy"] = state?.forceDestroy;
            resourceInputs["identifier"] = state?.identifier;
            resourceInputs["identifierPrefix"] = state?.identifierPrefix;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.definitions === undefined && !opts.urn) {
                throw new Error("Missing required property 'definitions'");
            }
            resourceInputs["definitions"] = args?.definitions;
            resourceInputs["description"] = args?.description;
            resourceInputs["forceDestroy"] = args?.forceDestroy;
            resourceInputs["identifier"] = args?.identifier;
            resourceInputs["identifierPrefix"] = args?.identifierPrefix;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SnapshotSchedule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SnapshotSchedule = SnapshotSchedule;
/** @internal */
SnapshotSchedule.__pulumiType = 'aws:redshift/snapshotSchedule:SnapshotSchedule';
