// @pulumi/aws@7.7.0/kms/replicaKey.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReplicaKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages a KMS multi-Region replica key.
 *
 * ## Example Usage
 *
 * ### AWS Provider v6 (and below)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const primary = new aws.kms.Key("primary", {
 *     description: "Multi-Region primary key",
 *     deletionWindowInDays: 30,
 *     multiRegion: true,
 * });
 * const replica = new aws.kms.ReplicaKey("replica", {
 *     description: "Multi-Region replica key",
 *     deletionWindowInDays: 7,
 *     primaryKeyArn: primary.arn,
 * });
 * ```
 *
 * ### AWS Provider v7 (and above)
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const primary = new aws.kms.Key("primary", {
 *     region: "us-east-1",
 *     description: "Multi-Region primary key",
 *     deletionWindowInDays: 30,
 *     multiRegion: true,
 * });
 * const replica = new aws.kms.ReplicaKey("replica", {
 *     description: "Multi-Region replica key",
 *     deletionWindowInDays: 7,
 *     primaryKeyArn: primary.arn,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import KMS multi-Region replica keys using the `id`. For example:
 *
 * ```sh
 * $ pulumi import aws:kms/replicaKey:ReplicaKey example 1234abcd-12ab-34cd-56ef-1234567890ab
 * ```
 */
class ReplicaKey extends pulumi.CustomResource {
    /**
     * Get an existing ReplicaKey resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ReplicaKey(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ReplicaKey.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ReplicaKey.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["bypassPolicyLockoutSafetyCheck"] = state?.bypassPolicyLockoutSafetyCheck;
            resourceInputs["deletionWindowInDays"] = state?.deletionWindowInDays;
            resourceInputs["description"] = state?.description;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["keyId"] = state?.keyId;
            resourceInputs["keyRotationEnabled"] = state?.keyRotationEnabled;
            resourceInputs["keySpec"] = state?.keySpec;
            resourceInputs["keyUsage"] = state?.keyUsage;
            resourceInputs["policy"] = state?.policy;
            resourceInputs["primaryKeyArn"] = state?.primaryKeyArn;
            resourceInputs["region"] = state?.region;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
        }
        else {
            const args = argsOrState;
            if (args?.primaryKeyArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'primaryKeyArn'");
            }
            resourceInputs["bypassPolicyLockoutSafetyCheck"] = args?.bypassPolicyLockoutSafetyCheck;
            resourceInputs["deletionWindowInDays"] = args?.deletionWindowInDays;
            resourceInputs["description"] = args?.description;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["policy"] = args?.policy;
            resourceInputs["primaryKeyArn"] = args?.primaryKeyArn;
            resourceInputs["region"] = args?.region;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["keyId"] = undefined /*out*/;
            resourceInputs["keyRotationEnabled"] = undefined /*out*/;
            resourceInputs["keySpec"] = undefined /*out*/;
            resourceInputs["keyUsage"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ReplicaKey.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ReplicaKey = ReplicaKey;
/** @internal */
ReplicaKey.__pulumiType = 'aws:kms/replicaKey:ReplicaKey';
