// @pulumi/aws@7.7.0/storagegateway/storedIscsiVolume.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoredIscsiVolume = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Manages an AWS Storage Gateway stored iSCSI volume.
 *
 * > **NOTE:** The gateway must have a working storage added (e.g., via the `aws.storagegateway.WorkingStorage` resource) before the volume is operational to clients, however the Storage Gateway API will allow volume creation without error in that case and return volume status as `WORKING STORAGE NOT CONFIGURED`.
 *
 * ## Example Usage
 *
 * ### Create Empty Stored iSCSI Volume
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.storagegateway.StoredIscsiVolume("example", {
 *     gatewayArn: exampleAwsStoragegatewayCache.gatewayArn,
 *     networkInterfaceId: exampleAwsInstance.privateIp,
 *     targetName: "example",
 *     preserveExistingData: false,
 *     diskId: test.id,
 * });
 * ```
 *
 * ### Create Stored iSCSI Volume From Snapshot
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as aws from "@pulumi/aws";
 *
 * const example = new aws.storagegateway.StoredIscsiVolume("example", {
 *     gatewayArn: exampleAwsStoragegatewayCache.gatewayArn,
 *     networkInterfaceId: exampleAwsInstance.privateIp,
 *     snapshotId: exampleAwsEbsSnapshot.id,
 *     targetName: "example",
 *     preserveExistingData: false,
 *     diskId: test.id,
 * });
 * ```
 *
 * ## Import
 *
 * Using `pulumi import`, import `aws_storagegateway_stored_iscsi_volume` using the volume Amazon Resource Name (ARN). For example:
 *
 * ```sh
 * $ pulumi import aws:storagegateway/storedIscsiVolume:StoredIscsiVolume example arn:aws:storagegateway:us-east-1:123456789012:gateway/sgw-12345678/volume/vol-12345678
 * ```
 */
class StoredIscsiVolume extends pulumi.CustomResource {
    /**
     * Get an existing StoredIscsiVolume resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StoredIscsiVolume(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of StoredIscsiVolume.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StoredIscsiVolume.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["arn"] = state?.arn;
            resourceInputs["chapEnabled"] = state?.chapEnabled;
            resourceInputs["diskId"] = state?.diskId;
            resourceInputs["gatewayArn"] = state?.gatewayArn;
            resourceInputs["kmsEncrypted"] = state?.kmsEncrypted;
            resourceInputs["kmsKey"] = state?.kmsKey;
            resourceInputs["lunNumber"] = state?.lunNumber;
            resourceInputs["networkInterfaceId"] = state?.networkInterfaceId;
            resourceInputs["networkInterfacePort"] = state?.networkInterfacePort;
            resourceInputs["preserveExistingData"] = state?.preserveExistingData;
            resourceInputs["region"] = state?.region;
            resourceInputs["snapshotId"] = state?.snapshotId;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tagsAll"] = state?.tagsAll;
            resourceInputs["targetArn"] = state?.targetArn;
            resourceInputs["targetName"] = state?.targetName;
            resourceInputs["volumeAttachmentStatus"] = state?.volumeAttachmentStatus;
            resourceInputs["volumeId"] = state?.volumeId;
            resourceInputs["volumeSizeInBytes"] = state?.volumeSizeInBytes;
            resourceInputs["volumeStatus"] = state?.volumeStatus;
            resourceInputs["volumeType"] = state?.volumeType;
        }
        else {
            const args = argsOrState;
            if (args?.diskId === undefined && !opts.urn) {
                throw new Error("Missing required property 'diskId'");
            }
            if (args?.gatewayArn === undefined && !opts.urn) {
                throw new Error("Missing required property 'gatewayArn'");
            }
            if (args?.networkInterfaceId === undefined && !opts.urn) {
                throw new Error("Missing required property 'networkInterfaceId'");
            }
            if (args?.preserveExistingData === undefined && !opts.urn) {
                throw new Error("Missing required property 'preserveExistingData'");
            }
            if (args?.targetName === undefined && !opts.urn) {
                throw new Error("Missing required property 'targetName'");
            }
            resourceInputs["diskId"] = args?.diskId;
            resourceInputs["gatewayArn"] = args?.gatewayArn;
            resourceInputs["kmsEncrypted"] = args?.kmsEncrypted;
            resourceInputs["kmsKey"] = args?.kmsKey;
            resourceInputs["networkInterfaceId"] = args?.networkInterfaceId;
            resourceInputs["preserveExistingData"] = args?.preserveExistingData;
            resourceInputs["region"] = args?.region;
            resourceInputs["snapshotId"] = args?.snapshotId;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["targetName"] = args?.targetName;
            resourceInputs["arn"] = undefined /*out*/;
            resourceInputs["chapEnabled"] = undefined /*out*/;
            resourceInputs["lunNumber"] = undefined /*out*/;
            resourceInputs["networkInterfacePort"] = undefined /*out*/;
            resourceInputs["tagsAll"] = undefined /*out*/;
            resourceInputs["targetArn"] = undefined /*out*/;
            resourceInputs["volumeAttachmentStatus"] = undefined /*out*/;
            resourceInputs["volumeId"] = undefined /*out*/;
            resourceInputs["volumeSizeInBytes"] = undefined /*out*/;
            resourceInputs["volumeStatus"] = undefined /*out*/;
            resourceInputs["volumeType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StoredIscsiVolume.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StoredIscsiVolume = StoredIscsiVolume;
/** @internal */
StoredIscsiVolume.__pulumiType = 'aws:storagegateway/storedIscsiVolume:StoredIscsiVolume';
